// vocs/_lib/vite/.vocs/dist/assets/android_support-BcPYdecI.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Android Support",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "android-support",
        children: ["Android Support", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#android-support",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx("aside", {
      children: jsx(_components.p, {
        children: "⚠️ Currently ZCAM is only supported on iOS. Below is an exploration into current C2PA support for Android and a high-level implementation plan for Android ZCAM."
      })
    }), "\n", jsx(_components.p, {
      children: "Currently, only two Android phone companies support C2PA: Google with their latest Pixel 10 and Samsung."
    }), "\n", jsx(_components.p, {
      children: "Google's Pixel 10 has full C2PA support for photos captured on the camera. The design is quite similar to ZCAM's where they use a key generated in the secure enclave to sign the manifest at capture time."
    }), "\n", jsx(_components.p, {
      children: "Samsung's C2PA implementation is less useful. It only supports attaching C2PA manifests to photos that were generated using AI."
    }), "\n", jsx(_components.p, {
      children: "On the other hand, Android phones (specifically those that support the Google Play Store) are all equipped with the necessary primitives to support a ZCAM flow. Note that all Androids that support the Google Play Store are required to have some secure enclave support. On the iOS side, ZCAM leverages the Apple App Attest API which provides the following guarantees:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "A key was generated on the secure enclave and is scoped to the calling app"
      }), "\n", jsx(_components.li, {
        children: "The calling app itself is non-tampered and the binary is trustworthy"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Android separates these functionalities into two independent APIs:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Android Key Attestation"
        }), ": API to obtain an attestation that a given key used by an app is indeed hardware-backed"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Google Play Integrity API"
        }), ": API used to ensure a given request is from a valid install of an app"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: 'By integrating these two APIs in the Android implementation, we will get the same "coverage" that the Apple App Attest API guarantees. The rest of the ZCAM protocol flows as intended, using C2PA and SP1 as necessary.'
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
