// vocs/_lib/vite/.vocs/dist/assets/architecture-CbPkURbm.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Architecture",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "architecture",
        children: ["Architecture", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#architecture",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "overview",
      children: ["Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#overview",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM SDK allows an integrating app to:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Take a picture"
      }), "\n", jsx(_components.li, {
        children: "Generate and sign a valid C2PA manifest over the photo (including a signature using the device key) including bindings to generate an SP1 proof"
      }), "\n", jsx(_components.li, {
        children: "Generate a verifiable SP1 proof that the signature is valid, was signed using a key bound to the app, and corresponds to the photo, attaching that to a valid C2PA manifest"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "The core flow uses cryptographic keys stored on the device's Secure Enclave to sign and attest that a photo was taken using the device's camera and has not been tampered with."
    }), "\n", jsxs(_components.h2, {
      id: "cryptographic-keys",
      children: ["Cryptographic Keys", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#cryptographic-keys",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM architecture generates and leverages two keys:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Key"
          }), jsx(_components.th, {
            children: "Purpose"
          }), jsx(_components.th, {
            children: "Apple Attested"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Device Key"
            })
          }), jsx(_components.td, {
            children: "Signs over a hash of the photo"
          }), jsx(_components.td, {
            children: "Yes"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Content Key"
            })
          }), jsx(_components.td, {
            children: "Signs the C2PA manifest"
          }), jsx(_components.td, {
            children: "No"
          })]
        })]
      })]
    }), "\n", jsx(_components.p, {
      children: "Both keys live in the Secure Enclave of the iPhone device. Importantly, the device key is specifically attested to have been generated on the Secure Enclave and is scoped to the specific app."
    }), "\n", jsxs(_components.h2, {
      id: "photo-lifecycle",
      children: ["Photo Lifecycle", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#photo-lifecycle",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "A photo progresses through three states to become a verified zphoto:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "State"
          }), jsx(_components.th, {
            children: "C2PA Assertion"
          }), jsx(_components.th, {
            children: "Description"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Raw"
            })
          }), jsx(_components.td, {
            children: "None"
          }), jsx(_components.td, {
            children: "Original photo taken by phone, including metadata. No verifiable, attested traits."
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Provable"
            })
          }), jsx(_components.td, {
            children: jsx(_components.code, {
              children: "succinct.bindings"
            })
          }), jsx(_components.td, {
            children: "Embeds a C2PA manifest including required inputs for generating a zero knowledge proof. This includes a signature using an Apple attested key."
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Verified (zphoto)"
            })
          }), jsx(_components.td, {
            children: jsx(_components.code, {
              children: "succinct.proof"
            })
          }), jsx(_components.td, {
            children: "Embeds a C2PA manifest including a verifiable zero knowledge proof that the photo was signed using a valid Apple attested key."
          })]
        })]
      })]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "Raw Photo → ["
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                textDecoration: "underline",
                "--shiki-dark": "#96D0FF",
                "--shiki-dark-text-decoration": "inherit"
              },
              children: "Capture"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "] → Provable Photo → ["
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                textDecoration: "underline",
                "--shiki-dark": "#96D0FF",
                "--shiki-dark-text-decoration": "inherit"
              },
              children: "Prove"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "] → Verified zphoto"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "                         (bindings)                    (proof)"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "flow",
      children: ["Flow", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#flow",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "1-initialization",
      children: ["1. Initialization", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#1-initialization",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The SDK first gets initialized. The main logic here is to generate or load the two necessary keys (device key and content key)."
    }), "\n", jsxs(_components.h3, {
      id: "2-capture",
      children: ["2. Capture", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#2-capture",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["The capture step allows a user to take a photo and output a ", jsx(_components.em, {
        children: "provable"
      }), " photo."]
    }), "\n", jsx(_components.p, {
      children: "When a user wants to take a zphoto, the SDK opens the camera screen. After the photo is taken, the SDK embeds a valid C2PA manifest by doing the following:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The photo is hashed, then signed using the device key. This outputs an Apple Attest assertion including a signature over the hash."
      }), "\n", jsxs(_components.li, {
        children: ["The C2PA manifest is initialized with a ", jsx(_components.em, {
          children: "capture"
        }), " action including metadata: time of photo, OS, device model, software version, etc."]
      }), "\n", jsxs(_components.li, {
        children: ["A ", jsx(_components.code, {
          children: "succinct.bindings"
        }), " assertion is added to the manifest including data necessary to prove that the photo was signed using a valid Apple attested key."]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/diagram_capture.png",
        alt: "Capture flow diagram"
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Output:"
      }), " A provable photo with a valid C2PA manifest containing metadata and an assertion that the photo was signed with an Apple attested key."]
    }), "\n", jsxs(_components.h3, {
      id: "3-prove",
      children: ["3. Prove", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#3-prove",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["The prove step takes the provable photo with C2PA manifest including a ", jsx(_components.code, {
        children: "succinct.bindings"
      }), " assertion and generates a proof verifying its validity. This is done by using the SP1 Prover Network to prove the ZCAM program with the photo (and C2PA manifest) as input."]
    }), "\n", jsx(_components.p, {
      children: "The SDK receives the proof from the network, then updates the photo and C2PA manifest:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["The ", jsx(_components.code, {
          children: "succinct.bindings"
        }), " assertion is removed"]
      }), "\n", jsxs(_components.li, {
        children: ["A ", jsx(_components.code, {
          children: "succinct.proof"
        }), " assertion is added containing the verifiable proof"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/diagram_prove.png",
        alt: "Capture flow diagram"
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Output:"
      }), " The final zphoto with a valid C2PA manifest including a verifiable proof."]
    }), "\n", jsxs(_components.h3, {
      id: "4-verify",
      children: ["4. Verify", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#4-verify",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The SDK includes logic for parsing and verifying the proof from the C2PA manifest:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Extract the manifest from the photo"
      }), "\n", jsx(_components.li, {
        children: "Verify the proof in the manifest"
      }), "\n", jsx(_components.li, {
        children: "Verify the photo hash matches the manifest"
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
