// vocs/_lib/vite/.vocs/dist/assets/background-DfkQNfy0.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Background",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "background",
        children: ["Background", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#background",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "c2pa",
      children: ["C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["C2PA is an open source standard that allows parties to attach metadata that is ", jsx(_components.em, {
        children: "cryptographically bound"
      }), " to a specified image. This is done using a ", jsx(_components.em, {
        children: "C2PA Manifest"
      }), " which is tied to a specific image, including details and metadata about said image and also a cryptographic signature over these details."]
    }), "\n", jsx(_components.p, {
      children: "The manifest can hold any information about the photo including metadata about what device it was taken on, size of the image etc. The key point is that it is signed using some key that attests to the accuracy of the associated information."
    }), "\n", jsx(_components.p, {
      children: "The manifest is then embedded directly into the photo file. The manner in which this is done depends. For example, it can be embedded into the metadata section of the JPEG file, or can be embedded as an invisible watermark encoding the manifest data."
    }), "\n", jsx(_components.p, {
      children: "ZCAM leverages the C2PA standard to embed a verifiable zero knowledge proof that the photo was taken on an iPhone and signed using a secure enclave private key that only the ZCAM SDK can use."
    }), "\n", jsxs(_components.h2, {
      id: "apple-app-attest",
      children: ["Apple App Attest", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#apple-app-attest",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["ZCAM uses Apple's App Attest service to ensure that a ZPhoto was indeed ", jsx(_components.em, {
        children: "signed"
      }), " by the ZCAM SDK."]
    }), "\n", jsxs(_components.p, {
      children: ["The ", jsx(_components.em, {
        children: "intended"
      }), " purpose of App Attest is to ensure requests coming from an app to a backend service is actually from the intended app. i.e. It guarantees that a backend service knows that it's only servicing requests from the associated mobile app."]
    }), "\n", jsxs(_components.p, {
      children: ["At a high level, this is done by generating a new signing keypair on the device's secure enclave ", jsx(_components.em, {
        children: "that only the specific mobile app can access"
      }), ". There is then a protocol for Apple to ", jsx(_components.em, {
        children: "attest"
      }), " to the keypair as having been generated on said mobile app. This ensures any subsequent signatures using the keypair is guaranteed to be from the app."]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/diagram_appattest.png",
        alt: "App Attest protocol diagram"
      })
    }), "\n", jsx(_components.p, {
      children: "Specifically, the App Attest protocol provides an attestation from Apple's backend server that is bound to some unique nonce and the public key of the secure enclave keypair. The app can then send this attestation as part of a request, such that the receiving backend service can verify the attestation."
    }), "\n", jsx(_components.p, {
      children: "Note, ZCAM leverages Apple App Attest, but uses it in a subtle but different way. There are two key differences:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The unique nonce used in the attestation is actually derived on the app itself, instead of receiving it from some backend service."
      }), "\n", jsx(_components.li, {
        children: "The attestation is verified in a zero knowledge proof instead of by a backend service."
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "ios-secure-enclave",
      children: ["iOS Secure Enclave", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ios-secure-enclave",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "TODO"
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
