// vocs/_lib/vite/.vocs/dist/assets/comparisons_to_other_approaches-BokteJUC.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "ZCAM vs Other Approaches",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "zcam-vs-other-approaches",
        children: ["ZCAM vs Other Approaches", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#zcam-vs-other-approaches",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "There already exist numerous technologies in the authenticity and verifiability of images space. These solutions, however, solve for slightly different problem spaces and provide varying levels of functionality."
    }), "\n", jsx(_components.aside, {
      "data-callout": "info",
      children: jsx(_components.p, {
        children: "ZCAM provides a complete solution for mobile photo authenticity verification. It combines hardware-backed attestation, optional zero-knowledge proofs, embedded verification data, and simple verification output (verified/not verified) into a single integrated system designed specifically for human-taken photos on mobile devices."
      })
    }), "\n", jsx(_components.p, {
      children: "Below is a feature comparison showing what each approach provides. See further sections below for more details on other tools and advantages that ZCAM can provide."
    }), "\n", jsxs(_components.h2, {
      id: "comparison-table",
      children: ["Comparison Table", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#comparison-table",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Approach"
          }), jsx(_components.th, {
            children: "Hardware Attestation"
          }), jsx(_components.th, {
            children: "ZK Proofs"
          }), jsx(_components.th, {
            children: "Embedded Verification"
          }), jsx(_components.th, {
            children: "User-friendly Output"
          }), jsx(_components.th, {
            children: "Camera Photos"
          }), jsx(_components.th, {
            children: "Use Case"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "ZCAM"
            })
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Complete mobile photo authenticity solution"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "C2PA (generic)"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Edit history tracking, content provenance"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Physical Camera"
            })
          }), jsx(_components.td, {
            children: "❓"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "For C2PA-compatible cameras, note only some cameras support hardware-backed signing"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "JPEG Trust"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Trust evaluation framework (leverages C2PA)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "SynthID"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "AI watermarking and detection"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "AI Content Marking"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "AI-generated content marking (watermarking or C2PA)"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "other-tooling",
      children: ["Other tooling", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#other-tooling",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "c2pa",
      children: ["C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["The C2PA content credential standard aims to create a ", jsx(_components.em, {
        children: "verifiable history"
      }), " of a given image."]
    }), "\n", jsxs(_components.p, {
      children: ["Note, it is ", jsx(_components.em, {
        children: "not"
      }), ' specific to "authentic" images, i.e. it can be applied to both camera-captured pictures, as well as AI-generated images. It creates a specification for attaching contextual metadata as an image evolves over time. For example, an initial image can be captured using a camera or generated using AI. An initial ', jsx(_components.em, {
        children: "C2PA Manifest"
      }), " is attached to the image indicating its origin. Thereafter, as edits are made (using C2PA-compatible editors), new manifests are attached indicating the edits made (e.g. cropping, brightness balancing, etc.). All of these manifests are signed using public key cryptography."]
    }), "\n", jsx(_components.strong, {
      children: "Key Differences from ZCAM:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "C2PA doesn't require hardware attestation. any signing key can be used to sign attestations."
      }), "\n", jsx(_components.li, {
        children: "C2PA focuses on edit history, while ZCAM focuses on capture authenticity"
      }), "\n", jsx(_components.li, {
        children: "ZCAM uses C2PA as the container format but adds hardware-backed guarantees"
      }), "\n", jsx(_components.li, {
        children: "C2PA is a more flexible, generic tooling for attaching historical metadata to an image, ZCAM focuses on proving authenticity of an iamge."
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["More technical information can be found in the ", jsx(_components.a, {
        href: "/overview/how_it_works",
        children: "How it Works"
      }), " section under C2PA."]
    }), "\n", jsxs(_components.h3, {
      id: "jpeg-trust",
      children: ["JPEG Trust", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#jpeg-trust",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["JPEG Trust provides a framework for evaluating the trustworthiness of an image. It is actually compatible with C2PA, and relies ", jsx(_components.em, {
        children: "on"
      }), " the data encoded in C2PA manifests."]
    }), "\n", jsxs(_components.p, {
      children: ["The main value here is the ability to create trust ", jsx(_components.em, {
        children: "profiles"
      }), " and ", jsx(_components.em, {
        children: "evaluators"
      }), " that leverage the bound data, which allows users to conclude the authenticity for a given photo."]
    }), "\n", jsx(_components.strong, {
      children: "Key Differences from ZCAM:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "JPEG Trust is an evaluation framework, not a capture/verification system"
      }), "\n", jsx(_components.li, {
        children: "It can work with ZCAM-generated C2PA manifests"
      }), "\n", jsx(_components.li, {
        children: "Focuses on trust scoring rather than cryptographic verification"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "synthid",
      children: ["SynthID", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#synthid",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "SynthID, developed by Google, is an invisible watermark framework that can embed data into an image in a more resilient format. Solutions that use the JUMBF metadata are susceptible to getting the metadata stripped, the photo getting edited/cropped, etc. The actual embedded data is based on an AI model, and can later be read and analyzed by another model that decides whether a photo is AI-generated or not."
    }), "\n", jsx(_components.strong, {
      children: "Key Differences from ZCAM:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "SynthID focuses on AI detection, ZCAM focuses on authenticity verification"
      }), "\n", jsx(_components.li, {
        children: "ZCAM provides cryptographic guarantees, SynthID provides probabilistic detection"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "iptc-photo-metadata",
      children: ["IPTC Photo Metadata", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#iptc-photo-metadata",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The IPTC Photo Metadata simply defines standardized nomenclature and labeling for the aforementioned protocols. It's a metadata standard rather than a verification system."
    }), "\n", jsx(_components.strong, {
      children: "Key Differences from ZCAM:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "IPTC is a metadata format, not a verification system"
      }), "\n", jsx(_components.li, {
        children: "Can be used alongside ZCAM for standardized metadata fields"
      }), "\n", jsx(_components.li, {
        children: "No cryptographic guarantees or hardware attestation"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "ai-content-marking",
      children: ["AI Content Marking", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ai-content-marking",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Several companies provide solutions for marking AI-generated content, using different approaches:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Meta Watermarking"
        }), ": Meta adds watermarks specifically to AI-generated content. These watermarks are meant to be durable, handling cropping and edit actions, etc."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "OpenAI C2PA"
        }), ": OpenAI (e.g. ChatGPT) generated images include C2PA manifests indicating they were generated using DALL-E and on ChatGPT. This is also included in Azure's OpenAI service."]
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Key Differences from ZCAM:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "These solutions focus on marking AI-generated content, while ZCAM verifies authentic human-captured photos"
      }), "\n", jsx(_components.li, {
        children: "No hardware attestation in these implementations"
      }), "\n", jsx(_components.li, {
        children: "Different goals: provenance/marking for AI content vs. authenticity verification for human-captured photos"
      }), "\n", jsx(_components.li, {
        children: "Watermark-based (Meta) or C2PA-based (OpenAI) rather than hardware-backed cryptographic verification"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "when-to-use-zcam",
      children: ["When to Use ZCAM", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#when-to-use-zcam",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM is the right choice when you need:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Hardware-backed authenticity"
        }), ": Guarantees that photos were taken on a specific device"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Cryptographic verification"
        }), ": Strong cryptographic proofs rather than probabilistic detection"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Mobile-first"
        }), ": Designed specifically for iOS (and planned Android) mobile apps"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Simple verification"
        }), ": Easy-to-use API that returns verified/not verified without requiring manifest parsing"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Complete solution"
        }), ": End-to-end flow from capture to verification with optional ZK proof integration"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Consider other approaches when:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "You need to mark AI-generated content (use SynthID or AI Content Marking solutions)"
      }), "\n", jsx(_components.li, {
        children: "You want a general-purpose edit history system (use generic C2PA)"
      }), "\n", jsx(_components.li, {
        children: "You need watermarking resilient to heavy editing (use SynthID)"
      }), "\n", jsx(_components.li, {
        children: "You're working with non-mobile platforms (use C2PA or other standards)"
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
