// vocs/_lib/vite/.vocs/dist/assets/current_state_of_the_art-DlrbhoUo.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Current State of the Art",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    header: "header",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "current-state-of-the-art",
        children: ["Current State of the Art", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#current-state-of-the-art",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: 'There already exist numerous technologies in the authenticity and "verifiable history of an image" space. All these solutions, however, solve for slightly different problem sets. Below is a brief introduction to the space:'
    }), "\n", jsxs(_components.h2, {
      id: "c2pa",
      children: ["C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["The C2PA content credential standard aims to create a ", jsx(_components.em, {
        children: "verifiable history"
      }), " of a given image. Note, it is ", jsx(_components.em, {
        children: "not"
      }), ' specific to "authentic" images, i.e. it can be applied to both camera-captured pictures, as well as AI-generated images. It creates a specification for attaching contextual metadata as an image evolves over time. For example, an initial image can be captured using a camera or generated using AI. An initial ', jsx(_components.em, {
        children: "C2PA Manifest"
      }), ' is attached to the image indicating its origin. Thereafter, as edits are made (using C2PA-compatible editors), new manifests are attached indicating the edits made (e.g. cropping, brightness balancing, etc.). All of these manifests are signed using public key cryptography. More technical information can be found in the "Background" section under C2PA.']
    }), "\n", jsxs(_components.h2, {
      id: "jpeg-trust",
      children: ["JPEG Trust", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#jpeg-trust",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["JPEG Trust provides a framework for evaluating the trustworthiness of an image. It is actually compatible with C2PA, and relies ", jsx(_components.em, {
        children: "on"
      }), " the data encoded in C2PA manifests. The main value here is the ability to create trust ", jsx(_components.em, {
        children: "profiles"
      }), " and ", jsx(_components.em, {
        children: "evaluators"
      }), " that leverage the bound data, which allows users to conclude the authenticity for a given photo."]
    }), "\n", jsxs(_components.h2, {
      id: "synthid",
      children: ["SynthID", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#synthid",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "SynthID, developed by Google, is an invisible watermark framework that can embed data into an image in a more resilient format. Solutions that use the JUMBF metadata are susceptible to getting the metadata stripped, the photo getting edited/cropped, etc. The actual embedded data is based on an AI model, and can later be read and analyzed by another model that decides whether a photo is AI-generated or not."
    }), "\n", jsxs(_components.h2, {
      id: "iptc-photo-metadata",
      children: ["IPTC Photo Metadata", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#iptc-photo-metadata",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The IPTC Photo Metadata simply defines standardized nomenclature and labeling for the aforementioned protocols."
    }), "\n", jsxs(_components.h2, {
      id: "meta-watermarking",
      children: ["Meta Watermarking", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#meta-watermarking",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Meta has a similar technology to SynthID where they add watermarks specifically to ", jsx(_components.em, {
        children: "AI"
      }), "-generated content. These watermarks are also meant to be durable, handling cropping and edit actions, etc."]
    }), "\n", jsxs(_components.h2, {
      id: "openai-c2pa",
      children: ["OpenAI C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#openai-c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "OpenAI (e.g. ChatGPT) generated images include C2PA manifests indicating they were generated using DALL-E and on ChatGPT. This is also included in Azure's OpenAI service."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
