// vocs/_lib/vite/.vocs/dist/assets/editing_zcam_photos-hzig6BXS.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Editing a ZCAM Photo",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    header: "header",
    img: "img",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "editing-a-zcam-photo",
        children: ["Editing a ZCAM Photo", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#editing-a-zcam-photo",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "All ZCAM photos are C2PA compatible, meaning that any editing software that supports C2PA can be used to edit ZCAM photos while maintaining the original capture manifests."
    }), "\n", jsx(_components.p, {
      children: "The three main editing software options used today are: Capture One, Adobe Lightroom, and PhotoLab."
    }), "\n", jsxs(_components.p, {
      children: ["Currently, Capture One and Adobe Lightroom ", jsx(_components.em, {
        children: "support"
      }), " C2PA, while PhotoLab does ", jsx(_components.em, {
        children: "not"
      }), ". Generally, the flow is to include the content credentials on export."]
    }), "\n", jsxs(_components.h2, {
      id: "exporting-with-content-credentials",
      children: ["Exporting with Content Credentials", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#exporting-with-content-credentials",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When exporting a ZCAM photo from Capture One, you can include the C2PA content credentials in the export settings. This preserves the original capture manifest and adds an edit manifest indicating the changes made."
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/screenshot_captureone_c2pa.png",
        alt: "Exporting with content credentials in Capture One"
      })
    }), "\n", jsx(_components.p, {
      children: "After export, the resulting image will contain a C2PA manifest that includes both the original capture information and the edit history."
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/screenshot_manifest_captureone.png",
        alt: "C2PA manifest showing capture and edit history"
      })
    }), "\n", jsxs(_components.h2, {
      id: "zero-knowledge-proof-limitations",
      children: ["Zero Knowledge Proof Limitations", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#zero-knowledge-proof-limitations",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "On the other hand, the core SP1 zero knowledge proof will not be applicable after edits. This is because the proof takes in the photo bytes as input and hashes them. If the photo is edited, the hash will be different, causing the proof verification to fail. If the user wishes to edit the photo after the fact, they should use the non-zero knowledge flavor of ZCAM which verifies the capture and app attestation in the clear."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
