// vocs/_lib/vite/.vocs/dist/assets/editing_zcam_photos-CLZQk1w5.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Editing a ZCAM Photo",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "editing-a-zcam-photo",
        children: ["Editing a ZCAM Photo", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#editing-a-zcam-photo",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "All zPhotos captured with ZCAM are C2PA-compatible, so any C2PA-aware editor can preserve the original capture manifest while adding an edit manifest."
    }), "\n", jsx(_components.p, {
      children: "Common editing tools today include Capture One, Adobe Lightroom, and PhotoLab. Capture One and Adobe Lightroom support C2PA exports; PhotoLab does not. Typically editors include the content credentials on export."
    }), "\n", jsxs(_components.h2, {
      id: "exporting-with-content-credentials",
      children: ["Exporting with Content Credentials", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#exporting-with-content-credentials",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When exporting a zPhoto from Capture One, include the C2PA content credentials in the export settings. This preserves the original capture manifest and adds an edit manifest describing the changes."
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/screenshot_captureone_c2pa.png",
        alt: "Exporting with content credentials in Capture One"
      })
    }), "\n", jsx(_components.p, {
      children: "After export, the image will contain a C2PA manifest that includes both the original capture information and the edit history."
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/screenshot_manifest_captureone.png",
        alt: "C2PA manifest showing capture and edit history"
      })
    }), "\n", jsxs(_components.h2, {
      id: "verification-limitations-after-editing",
      children: ["Verification Limitations After Editing", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verification-limitations-after-editing",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Both bindings verification and ZK proof verification will fail if a zPhoto is edited. This happens because:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The Apple App Attest signature (used by bindings verification) covers the hash of the original image"
      }), "\n", jsx(_components.li, {
        children: "The ZK proof (used by proof verification) also attests to the original image hash"
      }), "\n", jsx(_components.li, {
        children: "Editing changes the image bytes and therefore the hash"
      }), "\n", jsx(_components.li, {
        children: "As a result, signature and proof verification fail when the hashes no longer match"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Important"
      }), ": Although C2PA edit manifests can preserve the original capture and add edit history, the cryptographic signature/proof tied to the original image will not verify against the edited bytes."]
    }), "\n", jsx(_components.p, {
      children: "If you must edit a zPhoto after capture, you need to generate a new signature or proof after editing; otherwise verification will fail."
    }), "\n", jsxs(_components.h3, {
      id: "future-work-zk-proofs-for-edit-chains",
      children: ["Future Work: ZK Proofs for Edit Chains", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#future-work-zk-proofs-for-edit-chains",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "In principle, zero-knowledge proofs could be extended to verify edit chains while preserving privacy. A ZK proof could assert that:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The original zPhoto was captured authentically"
      }), "\n", jsx(_components.li, {
        children: "Each edit in the chain was made or authorized by trusted editors"
      }), "\n", jsx(_components.li, {
        children: "The final image hash corresponds to the verified edit chain"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "This approach would require:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Proof generation after each edit (incurs computation cost)"
      }), "\n", jsx(_components.li, {
        children: "Including edit history in the proof circuit"
      }), "\n", jsx(_components.li, {
        children: "More complex SP1 programs to verify multiple C2PA manifests and their relationships"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "For now, users who need edits should either generate a new signature/proof after editing or accept that verification of the edited image will fail. The current implementation focuses on verifying the original capture, which provides the strongest guarantees for unedited zPhotos."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
