// vocs/_lib/vite/.vocs/dist/assets/editing_zcam_photos-D7NnGELh.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Editing a ZCAM Photo",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "editing-a-zcam-photo",
        children: ["Editing a ZCAM Photo", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#editing-a-zcam-photo",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "All ZCAM photos are C2PA compatible, meaning that any editing software that supports C2PA can be used to edit ZCAM photos while maintaining the original capture manifests."
    }), "\n", jsx(_components.p, {
      children: "The three main editing software options used today are: Capture One, Adobe Lightroom, and PhotoLab."
    }), "\n", jsxs(_components.p, {
      children: ["Currently, Capture One and Adobe Lightroom ", jsx(_components.em, {
        children: "support"
      }), " C2PA, while PhotoLab does ", jsx(_components.em, {
        children: "not"
      }), ". Generally, the flow is to include the content credentials on export."]
    }), "\n", jsxs(_components.h2, {
      id: "exporting-with-content-credentials",
      children: ["Exporting with Content Credentials", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#exporting-with-content-credentials",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When exporting a ZCAM photo from Capture One, you can include the C2PA content credentials in the export settings. This preserves the original capture manifest and adds an edit manifest indicating the changes made."
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/screenshot_captureone_c2pa.png",
        alt: "Exporting with content credentials in Capture One"
      })
    }), "\n", jsx(_components.p, {
      children: "After export, the resulting image will contain a C2PA manifest that includes both the original capture information and the edit history."
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/screenshot_manifest_captureone.png",
        alt: "C2PA manifest showing capture and edit history"
      })
    }), "\n", jsxs(_components.h2, {
      id: "verification-limitations-after-editing",
      children: ["Verification Limitations After Editing", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verification-limitations-after-editing",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Both bindings verification and ZK proof verification break when a photo is edited."
      }), " This is because:"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The Apple App Attest signature (used in bindings verification) is over the hash of the original image"
      }), "\n", jsx(_components.li, {
        children: "The ZK proof (used in proof verification) also verifies the original photo hash"
      }), "\n", jsx(_components.li, {
        children: "When a photo is edited, the image bytes change, causing the hash to change"
      }), "\n", jsx(_components.li, {
        children: "The signature/proof verification fails because the hash no longer matches"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Important"
      }), ": Even though C2PA edit manifests can preserve the original capture manifest and add edit history, the actual signature verification will fail because the signature is over the hash of the original image bytes, which no longer match the edited image."]
    }), "\n", jsx(_components.p, {
      children: "If you need to edit photos after capture, you must generate a new signature/proof after editing, or accept that verification will fail for edited photos."
    }), "\n", jsxs(_components.h3, {
      id: "future-zk-proofs-for-edit-chains",
      children: ["Future: ZK Proofs for Edit Chains", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#future-zk-proofs-for-edit-chains",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "In theory, zero-knowledge proofs could be extended to verify entire edit chains while maintaining privacy. A ZK proof could verify that:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The original photo was captured authentically"
      }), "\n", jsx(_components.li, {
        children: "Each edit in the chain was legitimate (signed by authorized editors)"
      }), "\n", jsx(_components.li, {
        children: "The final photo hash matches the edit chain"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "This would allow privacy-preserving verification even after edits, addressing concerns about maintaining authenticity through editing workflows (such as those used by news organizations like the BBC)."
    }), "\n", jsx(_components.p, {
      children: "However, this approach requires:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Proof generation after each edit"
        }), ": Each edit would need a new proof, adding computational cost"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Maintaining edit history in the proof"
        }), ": The proof circuit would need to verify the entire edit chain"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "More complex proof circuits"
        }), ": The SP1 program would need to verify multiple C2PA manifests and their relationships"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "For now, users who need to edit photos should:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Generate a ", jsx(_components.strong, {
          children: "new signature/proof"
        }), " after editing (this requires re-capturing or re-signing the edited image)"]
      }), "\n", jsx(_components.li, {
        children: "Or accept that verification will fail for edited photos (the original capture manifest and edit history will still be preserved in C2PA, but signature verification will fail)"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "The current implementation focuses on verifying the original capture, which provides the strongest guarantees for unedited photos."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
