// vocs/_lib/vite/.vocs/dist/assets/example_usecases-DdL0ZKds.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Example Use Cases",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "example-use-cases",
        children: ["Example Use Cases", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#example-use-cases",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.p, {
      children: ["Below are two real-world examples demonstrating how the ZCAM SDK can be integrated to solve different verification challenges. Each example highlights a distinct verification pattern: ", jsx(_components.strong, {
        children: "end-user verification"
      }), " (public-facing) versus ", jsx(_components.strong, {
        children: "internal verification"
      }), " (backend processing)."]
    }), "\n", jsxs(_components.aside, {
      "data-callout": "info",
      children: [jsxs(_components.p, {
        children: [jsx(_components.strong, {
          children: "Verification Patterns:"
        }), " While both use cases rely on verifiable photos, they differ in who performs the verification and where it happens:"]
      }), jsxs(_components.ul, {
        children: ["\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "End-user verification"
          }), ": Public-facing verification where readers/users verify photos themselves (e.g., news platform readers)"]
        }), "\n", jsxs(_components.li, {
          children: [jsx(_components.strong, {
            children: "Internal verification"
          }), ": Backend verification where the company verifies photos as part of their internal processes (e.g., insurance claim processing)"]
        }), "\n"]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "verifiable-news-platform",
      children: ["Verifiable News Platform", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verifiable-news-platform",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "use-case-overview",
      children: ["Use Case Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#use-case-overview",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "In today's media landscape, edited and AI-generated photos are a frequent concern for news consumers. A news platform hosting breaking news stories with field reporters may want to ensure their readers that all photos are verifiably authentic and unedited."
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "What We're Building:"
      }), " A complete verifiable journalism system consisting of:"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "A mobile camera app for field reporters to capture verifiable photos using ZCAM SDK"
      }), "\n", jsx(_components.li, {
        children: "Integration into the news platform's content management system to store and display verifiable photos"
      }), "\n", jsx(_components.li, {
        children: "Public-facing verification tools on the web platform allowing readers to independently verify photo authenticity"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Value of Verifiable Photos:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Trust & Credibility"
        }), ": Readers can independently verify that photos haven't been edited or manipulated, building trust in the platform's journalism"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "integration-architecture",
      children: ["Integration Architecture", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-architecture",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The news platform integrates ZCAM SDK in two places:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Mobile Camera App"
        }), " (for reporters): A lightweight iOS app that reporters use to capture verifiable photos"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Web Platform"
        }), " (for readers): Integration of verification functionality so readers can verify photos as authentic"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "mobile-camera-app-integration",
      children: ["Mobile Camera App Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mobile-camera-app-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The news platform builds a specialized camera app for reporters using the ZCAM SDK:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Installation"
        }), ": Follow the ", jsx(_components.a, {
          href: "/getting-started/installation",
          children: "Installation guide"
        }), " to set up the SDK"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Camera Setup"
        }), ": Use ", jsx(_components.code, {
          children: "react-native-zcam1-capture"
        }), " to initialize the camera:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "initCapture()"
            }), " to initialize device keys and attestation"]
          }), "\n", jsxs(_components.li, {
            children: ["Use the ", jsx(_components.code, {
              children: "<ZCamera>"
            }), " component to render the camera preview"]
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "takePhoto()"
            }), " to capture a photo with verifiable bindings"]
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Proof Generation"
        }), " (Optional): Use ", jsx(_components.code, {
          children: "react-native-zcam1-prove"
        }), " to generate zero-knowledge proofs:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Wrap your app with ", jsx(_components.code, {
              children: "ProverProvider"
            }), " and use the ", jsx(_components.code, {
              children: "useProver()"
            }), " hook"]
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "waitAndEmbedProof()"
            }), " to generate and embed the proof, or use ", jsx(_components.code, {
              children: "requestProof()"
            }), " + ", jsx(_components.code, {
              children: "useProofRequestStatus()"
            }), " for progress tracking"]
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Upload"
        }), ": Upload the verifiable photo to the news platform's backend for use in articles"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "web-platform-integration",
      children: ["Web Platform Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#web-platform-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "On the news platform's website, articles display verifiable photos with verification capabilities:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Photo Display"
        }), ": Photos embedded in articles include C2PA content credentials", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Users with C2PA browser extensions can automatically see content credentials"
          }), "\n", jsx(_components.li, {
            children: "The platform can display verification status badges"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Client-Side Verification"
        }), ": Integrate ", jsx(_components.code, {
          children: "react-native-zcam1-verify"
        }), " (or web equivalent) to enable verification:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: 'Add a "Verify Photo" button for each photo in articles'
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "verifyHash()"
            }), " and ", jsx(_components.code, {
              children: "verifyProof()"
            }), " methods to validate authenticity"]
          }), "\n", jsx(_components.li, {
            children: 'Display verification results to readers (e.g., "✓ Verified Authentic")'
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Metadata Display"
        }), ": Show readers metadata from the C2PA manifest:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Capture timestamp, device information, app identifier"
          }), "\n", jsx(_components.li, {
            children: "Allow readers to inspect the full verification chain"
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "car-insurance-claim-app",
      children: ["Car Insurance Claim App", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#car-insurance-claim-app",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "use-case-overview-1",
      children: ["Use Case Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#use-case-overview-1",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Most car insurance companies allow policyholders to file claims using photos from their mobile app. This has led to an increase in AI editing scams where claimants manipulate photos to file fraudulent claims. The ZCAM SDK ensures that photos uploaded for claims were taken on an iPhone using the insurance company's app, guaranteeing the photo hasn't been edited or altered since capture."
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "What We're Building:"
      }), " An automated fraud detection system integrated into the insurance claim workflow:"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Camera functionality within the existing insurance mobile app using ZCAM SDK for photo capture"
      }), "\n", jsx(_components.li, {
        children: "Backend verification service that automatically validates all submitted claim photos"
      }), "\n", jsx(_components.li, {
        children: "Automated triage system that flags suspicious claims based on verification results and metadata analysis"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Value of Verifiable Photos:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Fraud Prevention"
        }), ": Automatically detect and reject claims with manipulated or inauthentic photos, preventing fraudulent payouts"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Cost Reduction"
        }), ": Reduce manual review costs by automating verification and flagging only suspicious claims for human review"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Verification Pattern:"
      }), " Internal verification — the insurance company verifies photos as part of their backend claim processing workflow."]
    }), "\n", jsxs(_components.h3, {
      id: "integration-architecture-1",
      children: ["Integration Architecture", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#integration-architecture-1",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The insurance company integrates ZCAM SDK in two places:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Mobile App"
        }), " (for policyholders): Integration into the claim filing flow"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Backend Claim Processing"
        }), " (for insurance company): Automated verification during claim triage"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "mobile-app-integration",
      children: ["Mobile App Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mobile-app-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM SDK is integrated into the claim filing flow of the insurance app:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Installation"
        }), ": Follow the ", jsx(_components.a, {
          href: "/getting-started/installation",
          children: "Installation guide"
        }), " to set up the SDK"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Camera Integration"
        }), ": Inject the ", jsx(_components.code, {
          children: "<ZCamera>"
        }), " component into the claim flow where users must photograph their vehicle:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "The SDK provides camera preview functionality"
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "takePhoto()"
            }), " to capture a verifiable photo"]
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Proof Generation"
        }), " (Optional): Use ", jsx(_components.code, {
          children: "react-native-zcam1-prove"
        }), " to attach zero-knowledge proofs:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Use ", jsx(_components.code, {
              children: "waitAndEmbedProof()"
            }), " to generate and embed proofs"]
          }), "\n", jsx(_components.li, {
            children: "This provides additional cryptographic guarantees and privacy"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Claim Submission"
        }), ": When the user completes the claim flow, attach verifiable photos to the claim filing and upload to the insurance company's backend"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "backend-claim-processing-integration",
      children: ["Backend Claim Processing Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#backend-claim-processing-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "On the insurance company's backend, verification is integrated into the automated claim triage process:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Automated Verification"
        }), ": When a claim enters the triage flow, the system automatically verifies all attached photos:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Integrate ", jsx(_components.code, {
              children: "react-native-zcam1-verify"
            }), " (or server-side equivalent) into the backend"]
          }), "\n", jsxs(_components.li, {
            children: ["For each photo, call ", jsx(_components.code, {
              children: "verifyHash()"
            }), " and ", jsx(_components.code, {
              children: "verifyProof()"
            }), " methods"]
          }), "\n", jsx(_components.li, {
            children: "Check that the photo was taken using the insurance company's app"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Metadata Analysis"
        }), ": Leverage metadata from the C2PA manifest:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Verify location data matches the claim location"
          }), "\n", jsx(_components.li, {
            children: "Check timestamp aligns with the reported incident time"
          }), "\n", jsx(_components.li, {
            children: "Validate device and app information"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Claim Decision"
        }), ": Based on verification results:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: [jsx(_components.strong, {
              children: "Verification passes"
            }), ": Proceed with normal claim processing"]
          }), "\n", jsxs(_components.li, {
            children: [jsx(_components.strong, {
              children: "Verification fails"
            }), ": Flag claim for manual review or deny if photos are inauthentic"]
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Result"
      }), ": Insurance companies can automatically detect fraudulent claims with manipulated photos, reducing fraud and processing costs while maintaining a smooth experience for legitimate claimants."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
