// vocs/_lib/vite/.vocs/dist/assets/faq-JFWKhs3l.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "FAQs",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "faqs",
        children: ["FAQs", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#faqs",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "does-zcam-work-on-android",
      children: ["Does ZCAM work on Android?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#does-zcam-work-on-android",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Not at the moment, but we have an idea on how to implement it. See the ", jsx(_components.a, {
        href: "/technical-docs/android_support",
        children: "Android Support"
      }), " page for more details on our implementation plan."]
    }), "\n", jsxs(_components.h2, {
      id: "is-there-a-way-to-mimic-the-signature-from-the-enclave-is-it-100-impossible-or-just-unlikely-does-it-vary-by-os",
      children: ["Is there a way to mimic the signature from the enclave? Is it 100% impossible or just unlikely? Does it vary by OS?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#is-there-a-way-to-mimic-the-signature-from-the-enclave-is-it-100-impossible-or-just-unlikely-does-it-vary-by-os",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "With proper use of the App Attest (iOS) and Play Integrity/Key Attestation (Android) APIs, it should be 100% impossible to mimic a signature. This relies on Apple and Google ensuring these APIs work as intended, however."
    }), "\n", jsxs(_components.h2, {
      id: "have-we-considered-more-complicated-attack-vectors-such-as-jailbroken-devices--running-a-virtual-app-to-exploit-the-signatures",
      children: ['Have we considered more complicated attack vectors, such as jailbroken devices + running a "virtual app" to exploit the signatures?', jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#have-we-considered-more-complicated-attack-vectors-such-as-jailbroken-devices--running-a-virtual-app-to-exploit-the-signatures",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Yes, the main attack vector is being able to feed in photos that weren't actually captured from the camera into our ZCAM signing protocol. We can protect our SDK as much as possible and keep the integration as simple as possible, but there's really no guarantees to be had with jailbroken devices, etc. The third party is also responsible to ensure their integrating app doesn't have any glaring security gaps."
    }), "\n", jsxs(_components.h2, {
      id: "does-cropping-or-editing-break-the-photo-verification",
      children: ["Does cropping or editing break the photo verification?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#does-cropping-or-editing-break-the-photo-verification",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "If you try to remove the metadata or crop the photo, the photo will no longer be verifiable. We're looking into watermarking to insert data which could help with cropping, etc."
    }), "\n", jsxs(_components.h2, {
      id: "does-your-solution-detect-fakes-if-a-photo-is-taken-with-another-device",
      children: ["Does your solution detect fakes if a photo is taken with another device?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#does-your-solution-detect-fakes-if-a-photo-is-taken-with-another-device",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM verifies that a photo was taken using the specific device and app that generated the signature. If a photo is taken with a different device, it would not have the proper attestation and signatures, so it would fail verification."
    }), "\n", jsxs(_components.h2, {
      id: "can-it-detect-a-photo-of-a-photo",
      children: ["Can it detect a photo of a photo?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#can-it-detect-a-photo-of-a-photo",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: 'What we can do is provide "contextual" metadata, for example metadata around subject depth, brightness, etc. that would indicate a potential photo of a photo. However, this is not a primary feature of the core ZCAM verification system.'
    }), "\n", jsxs(_components.h2, {
      id: "have-you-built-any-other-features-on-top-of-the-signing--attestation-from-the-secure-enclave",
      children: ["Have you built any other features on top of the signing / attestation from the Secure Enclave?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#have-you-built-any-other-features-on-top-of-the-signing--attestation-from-the-secure-enclave",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM is our first mobile app built on top of Secure Enclave signing and attestation. The core functionality focuses on photo capture, signing, proof generation, and verification."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
