// vocs/_lib/vite/.vocs/dist/assets/faq-BPhs7I_K.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "FAQs",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "faqs",
        children: ["FAQs", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#faqs",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "android-support",
      children: ["Android Support", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#android-support",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "Does ZCAM work on Android?"
      })
    }), "\n", jsx(_components.p, {
      children: "Not at the moment, but we have a plan for Android implementation. Currently, only two Android phone companies support C2PA: Google with their latest Pixel 10 and Samsung."
    }), "\n", jsx(_components.p, {
      children: "Google's Pixel 10 has full C2PA support for photos captured on the camera. The design is quite similar to ZCAM's where they use a key generated in the secure enclave to sign the manifest at capture time."
    }), "\n", jsx(_components.p, {
      children: "Samsung's C2PA implementation is less useful. It only supports attaching C2PA manifests to photos that were generated using AI."
    }), "\n", jsx(_components.p, {
      children: "On the other hand, Android phones (specifically those that support the Google Play Store) are all equipped with the necessary primitives to support a ZCAM flow. Note that all Androids that support the Google Play Store are required to have some secure enclave support. On the iOS side, ZCAM leverages the Apple App Attest API which provides the following guarantees:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "A key was generated on the secure enclave and is scoped to the calling app"
      }), "\n", jsx(_components.li, {
        children: "The calling app itself is non-tampered and the binary is trustworthy"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Android separates these functionalities into two independent APIs:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Android Key Attestation"
        }), ": API to obtain an attestation that a given key used by an app is indeed hardware-backed"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Google Play Integrity API"
        }), ": API used to ensure a given request is from a valid install of an app"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: 'By integrating these two APIs in the Android implementation, we will get the same "coverage" that the Apple App Attest API guarantees. The rest of the ZCAM protocol flows as intended, using C2PA and SP1 as necessary.'
    }), "\n", jsxs(_components.h2, {
      id: "secure-enclave-guarantees",
      children: ["Secure Enclave Guarantees", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#secure-enclave-guarantees",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "Is there a way to mimic the signature from the enclave? Is it 100% impossible or just unlikely? Does it vary by OS?"
      })
    }), "\n", jsx(_components.p, {
      children: "With proper use of the App Attest (iOS) and Play Integrity/Key Attestation (Android) APIs, it should be 100% impossible to mimic a signature. This relies on Apple and Google ensuring these APIs work as intended, however."
    }), "\n", jsxs(_components.h2, {
      id: "jailbroken-devices",
      children: ["Jailbroken Devices", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#jailbroken-devices",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: 'Have we considered more complicated attack vectors, such as jailbroken devices + running a "virtual app" to exploit the signatures?'
      })
    }), "\n", jsx(_components.p, {
      children: "Yes, the main attack vector is being able to feed in photos that weren't actually captured from the camera into our ZCAM signing protocol. We can protect our SDK as much as possible and keep the integration as simple as possible, but there's really no guarantees to be had with jailbroken devices, etc. The third party is also responsible to ensure their integrating app doesn't have any glaring security gaps."
    }), "\n", jsxs(_components.h2, {
      id: "verification-post-edits",
      children: ["Verification Post-edits", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verification-post-edits",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "Does cropping or editing break the photo verification?"
      })
    }), "\n", jsxs(_components.p, {
      children: ["Yes, ", jsx(_components.strong, {
        children: "both bindings verification and ZK proof verification break when a photo is edited"
      }), ". This is because:"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The Apple App Attest signature (used in bindings verification) is over the hash of the original image"
      }), "\n", jsx(_components.li, {
        children: "The ZK proof (used in proof verification) also verifies the original photo hash"
      }), "\n", jsx(_components.li, {
        children: "When a photo is edited, the image bytes change, causing the hash to change"
      }), "\n", jsx(_components.li, {
        children: "The signature/proof verification fails because the hash no longer matches"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Even though C2PA edit manifests can preserve the original capture manifest and add edit history, the actual signature verification will fail because the signature is over the hash of the original image bytes."
    }), "\n", jsx(_components.p, {
      children: "If you remove the metadata entirely or crop in a way that strips the C2PA manifest, the photo will also no longer be verifiable. We're looking into watermarking to insert data which could help with cropping, etc."
    }), "\n", jsxs(_components.p, {
      children: ["See the ", jsx(_components.a, {
        href: "/c2pa-ecosystem/editing_zcam_photos",
        children: "Editing ZCAM Photos"
      }), " page for more details."]
    }), "\n", jsxs(_components.h2, {
      id: "cross-device-support",
      children: ["Cross Device Support", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#cross-device-support",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "Does your solution detect fakes if a photo is taken with another device?"
      })
    }), "\n", jsx(_components.p, {
      children: "ZCAM verifies that a photo was taken using the specific device and app that generated the signature. If a photo is taken with a different device, it would not have the proper attestation and signatures, so it would fail verification."
    }), "\n", jsxs(_components.h2, {
      id: "photo-of-photo",
      children: ["Photo of Photo", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#photo-of-photo",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "Can it detect a photo of a photo?"
      })
    }), "\n", jsx(_components.p, {
      children: 'What we can do is provide "contextual" metadata, for example metadata around subject depth, brightness, etc. that would indicate a potential photo of a photo. However, this is not a primary feature of the core ZCAM verification system.'
    }), "\n", jsxs(_components.h2, {
      id: "bindings-vs-proof-verification",
      children: ["Bindings vs Proof verification", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#bindings-vs-proof-verification",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "What's the difference between bindings verification and ZK proof verification?"
      })
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Bindings verification"
      }), " directly checks the Apple App Attest signature embedded in the C2PA manifest. It can be generated offline, however the attestation data is visible in the manifest."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "ZK proof verification"
      }), " generates a zero-knowledge proof that verifies the attestation without revealing sensitive details. It provides enhanced privacy but requires network access for proof generation."]
    }), "\n", jsx(_components.p, {
      children: "ZK proofs are completely optional. You can use ZCAM with bindings verification alone, which works entirely offline."
    }), "\n", jsx(_components.p, {
      children: "Choose ZK proof verification when:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Privacy is paramount—you want to verify authenticity without revealing attestation details"
      }), "\n", jsx(_components.li, {
        children: "You can tolerate network latency for proof generation"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["See the ", jsx(_components.a, {
        href: "/overview/how_it_works",
        children: "How it Works"
      }), " page for a detailed comparison and tradeoffs."]
    }), "\n", jsxs(_components.h2, {
      id: "metadata-stripping",
      children: ["Metadata Stripping", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#metadata-stripping",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.em, {
        children: "Which platforms preserve C2PA metadata when uploading images?"
      })
    }), "\n", jsx(_components.p, {
      children: 'Short answer, most platforms where a user uploads image files to share strip metadata. Platforms where the service is more to "host files" may keep the metadata. LinkedIn has first class support for C2PA.'
    }), "\n", jsxs("details", {
      children: [jsx("summary", {
        children: "Platform Compatibility Table"
      }), jsx(_components.p, {
        children: "This table documents how different platforms handle C2PA content credentials when images are uploaded and accessed through various methods:"
      }), jsxs(_components.table, {
        children: [jsx(_components.thead, {
          children: jsxs(_components.tr, {
            children: [jsx(_components.th, {
              children: jsx(_components.strong, {
                children: "Platform"
              })
            }), jsx(_components.th, {
              children: jsx(_components.strong, {
                children: "Upload Method"
              })
            }), jsx(_components.th, {
              children: jsx(_components.strong, {
                children: "Read Method"
              })
            }), jsx(_components.th, {
              children: jsx(_components.strong, {
                children: "Maintains Credential?"
              })
            }), jsx(_components.th, {
              children: jsx(_components.strong, {
                children: "Notes"
              })
            })]
          })
        }), jsxs(_components.tbody, {
          children: [jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Notion"
              })
            }), jsxs(_components.td, {
              children: ["Upload using ", jsx(_components.code, {
                children: "/file"
              })]
            }), jsxs(_components.td, {
              children: ["Click on link, hosted on ", jsx(_components.code, {
                children: "https://file.notion.so/..."
              })]
            }), jsx(_components.td, {
              children: "✅"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Notion"
              })
            }), jsxs(_components.td, {
              children: ["Upload using ", jsx(_components.code, {
                children: "/file"
              })]
            }), jsx(_components.td, {
              children: "Download"
            }), jsx(_components.td, {
              children: "✅"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Twitter"
              })
            }), jsx(_components.td, {
              children: "Upload using upload file functionality"
            }), jsx(_components.td, {
              children: "Download"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Twitter"
              })
            }), jsx(_components.td, {
              children: "Link to image"
            }), jsx(_components.td, {
              children: "Click on Link"
            }), jsx(_components.td, {
              children: "✅"
            }), jsx(_components.td, {
              children: "Doesn't render properly unless hosting has og tag"
            })]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Instagram"
              })
            }), jsx(_components.td, {
              children: "New post (ensure original ratio/ not cropped etc.)"
            }), jsx(_components.td, {
              children: "No way to download, can expect src to get uploaded image"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "LinkedIn"
              })
            }), jsx(_components.td, {
              children: "Upload in post"
            }), jsx(_components.td, {
              children: "Natively embedded"
            }), jsx(_components.td, {
              children: "✅"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "LinkedIn"
              })
            }), jsx(_components.td, {
              children: "Upload in post"
            }), jsx(_components.td, {
              children: "Download image"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Discord"
              })
            }), jsx(_components.td, {
              children: "Upload file"
            }), jsx(_components.td, {
              children: "Open hosted link"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {
              children: "Strips metadata"
            })]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Discord"
              })
            }), jsx(_components.td, {
              children: "Upload file"
            }), jsx(_components.td, {
              children: "Download file from hosted link"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Gmail"
              })
            }), jsx(_components.td, {
              children: "Attach file to email"
            }), jsx(_components.td, {
              children: "Click preview"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Gmail"
              })
            }), jsx(_components.td, {
              children: "Attach file to email"
            }), jsx(_components.td, {
              children: "Download"
            }), jsx(_components.td, {
              children: "✅"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Gmail"
              })
            }), jsx(_components.td, {
              children: "Embed file in email"
            }), jsx(_components.td, {
              children: "Download"
            }), jsx(_components.td, {
              children: "✅"
            }), jsx(_components.td, {})]
          }), jsxs(_components.tr, {
            children: [jsx(_components.td, {
              children: jsx(_components.strong, {
                children: "Gmail"
              })
            }), jsx(_components.td, {
              children: "Embed file in email"
            }), jsx(_components.td, {
              children: "Click preview"
            }), jsx(_components.td, {
              children: "❌"
            }), jsx(_components.td, {})]
          })]
        })]
      }), jsx(_components.p, {
        children: "Note: Cloudflare Images also supports C2PA."
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
