// vocs/_lib/vite/.vocs/dist/assets/hardware_camera_support-BVVxc8GM.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Hardware Camera Support",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    header: "header",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "hardware-camera-support",
        children: ["Hardware Camera Support", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#hardware-camera-support",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "We are beginning to see camera companies roll out C2PA-supported cameras. Leica was the first camera company to really lean into C2PA starting in late 2023, with other companies adding support since then."
    }), "\n", jsxs(_components.p, {
      children: ["It's worth noting that different companies have different levels of integration and security. For example, Leica uses a ", jsx(_components.em, {
        children: "hardware"
      }), "-backed signing system to sign the manifests, whereas other camera companies use a software-based solution. Obviously, the hardware-backed solutions will be harder to exploit or generate signatures over non-authentic pictures."]
    }), "\n", jsxs(_components.p, {
      children: ["Each camera company only has C2PA support for a ", jsx(_components.em, {
        children: "subset"
      }), " of their camera models. Generally speaking, their flagships seem most likely to have C2PA support."]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Brand"
            })
          }), jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Supported Models"
            })
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Leica"
            })
          }), jsx(_components.td, {
            children: "M11-P, M11-D, SL3-S"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Sony"
            })
          }), jsx(_components.td, {
            children: "Alpha 1, 1 II, 7 IV, 7S III, 9 III (plus video on FX3/FX30/PXW-Z300)"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Canon"
            })
          }), jsx(_components.td, {
            children: "EOS R1, EOS R5 Mark II"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Fujifilm"
            })
          }), jsx(_components.td, {
            children: "GFX100S II, X-T50"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Nikon"
            })
          }), jsx(_components.td, {
            children: "Z6 III (firmware + cloud cert)"
          })]
        })]
      })]
    }), "\n", jsxs(_components.p, {
      children: ["For more information, see: ", jsx(_components.a, {
        href: "https://yawnbox.eu/blog/c2pa-camera/",
        children: "https://yawnbox.eu/blog/c2pa-camera/"
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
