// vocs/_lib/vite/.vocs/dist/assets/how_it_works-BNzt8-cT.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "How it Works",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "how-it-works",
        children: ["How it Works", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#how-it-works",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "overview",
      children: ["Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#overview",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM enables photo authenticity verification through a combination of hardware-backed attestation, cryptographic signatures, and optional zero-knowledge proofs."
    }), "\n", jsx(_components.p, {
      children: "At a high level, ZCAM enables a user to take a photo, then leverages hardware-backed signing keys to sign the validity of the photo. This attestation is then attached to the photo using C2PA. Finally, the attestation can optionally be replaced with a zero knowledge proof that proves the verification of the attestation."
    }), "\n", jsx(_components.p, {
      children: "The system works in three main steps:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Capture"
        }), ": Take a photo and sign it with a hardware-backed key"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Prove"
        }), " (optional): Generate a zero-knowledge proof of the signature"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verify"
        }), ": Validate the photo's authenticity using either bindings or ZK proof verification"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "verification-modes",
      children: ["Verification Modes", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verification-modes",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM supports two verification modes:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Bindings verification"
        }), ": Verifies the Apple App Attest signature directly. This mode is faster and works offline. No zero-knowledge proofs are required."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "ZK proof verification"
        }), " (optional): Generates a zero-knowledge proof for enhanced privacy and verifiability. This provides additional cryptographic guarantees but requires network access for proof generation."]
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["The ZK proof step is ", jsx(_components.strong, {
        children: "optional"
      }), "—you can use ZCAM with bindings verification alone, or add ZK proofs for additional privacy guarantees."]
    }), "\n", jsxs(_components.h2, {
      id: "components",
      children: ["Components", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#components",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "This section describes the building blocks of ZCAM: Zero knowledge proofs generated using the SP1 zk virtual machine, C2PA for attaching metadata to an image and Apple's App attest service and secure enclave used for signing the validity of a photo."
    }), "\n", jsxs(_components.h3, {
      id: "zero-knowledge-proofs",
      children: ["Zero-Knowledge Proofs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#zero-knowledge-proofs",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Zero-knowledge proofs allow one party (the prover) to prove to another party (the verifier) that a statement is true without revealing any information beyond the validity of the statement itself."
    }), "\n", jsxs(_components.h3, {
      id: "sp1-zero-knowledge-virtual-machine",
      children: ["SP1 Zero-Knowledge Virtual Machine", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#sp1-zero-knowledge-virtual-machine",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["ZCAM uses ", jsx(_components.a, {
        href: "https://docs.succinct.xyz/docs/sp1/introduction",
        children: "SP1"
      }), ", Succinct's zero-knowledge virtual machine (zkVM). SP1 is a zkVM that proves the correct execution of programs compiled for the RISC-V architecture, meaning it can run and prove programs written in Rust, C++, C, or any language that compiles to RISC-V."]
    }), "\n", jsx(_components.p, {
      children: "Key benefits of SP1:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Maintainability"
        }), ": Write ZK programs in standard Rust without custom DSLs or complex circuits"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Faster Development"
        }), ": Skip months of low-level ZK engineering"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Performance"
        }), ": State-of-the-art proving speed and efficiency, benchmarked in production"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "In ZCAM, SP1 generates proofs that verify:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The photo was signed using a valid Apple App Attest key"
      }), "\n", jsx(_components.li, {
        children: "The signature corresponds to the photo hash"
      }), "\n", jsx(_components.li, {
        children: "The attestation chain is valid"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "These proofs are small, fast to verify, and don't reveal the underlying attestation data."
    }), "\n", jsx(_components.aside, {
      "data-callout": "info",
      children: jsxs(_components.p, {
        children: [jsx(_components.strong, {
          children: "ZK Proofs are Optional"
        }), ": You can use ZCAM with bindings verification alone. ZK proofs add privacy guarantees but require internet access to request proofs from the Succinct Prover Network. There is also higher latency given the time required for the network to prove the verification of an attestation."]
      })
    }), "\n", jsxs(_components.h3, {
      id: "c2pa",
      children: ["C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["C2PA is an open source standard that allows parties to attach metadata that is ", jsx(_components.em, {
        children: "cryptographically bound"
      }), " to a specified image. This is done using a ", jsx(_components.em, {
        children: "C2PA Manifest"
      }), " which is tied to a specific image, including details and metadata about said image and also a cryptographic signature over these details."]
    }), "\n", jsx(_components.p, {
      children: "The manifest can hold any information about the photo including metadata about what device it was taken on, size of the image etc. The key point is that it is signed using some key that attests to the accuracy of the associated information."
    }), "\n", jsx(_components.p, {
      children: "The manifest is then embedded directly into the photo file. The manner in which this is done depends. For example, it can be embedded into the metadata section of the JPEG file, or can be embedded as an invisible watermark encoding the manifest data."
    }), "\n", jsx(_components.p, {
      children: "ZCAM leverages the C2PA standard to embed verifiable attestation data (and optionally zero-knowledge proofs) that the photo was taken on an iPhone and signed using a secure enclave private key that only the ZCAM SDK can use."
    }), "\n", jsxs(_components.h3, {
      id: "apple-app-attest",
      children: ["Apple App Attest", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#apple-app-attest",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["ZCAM uses Apple's App Attest service to ensure that a ZPhoto was indeed ", jsx(_components.em, {
        children: "signed"
      }), " by the ZCAM SDK."]
    }), "\n", jsxs(_components.p, {
      children: ["The ", jsx(_components.em, {
        children: "intended"
      }), " purpose of App Attest is to ensure requests coming from an app to a backend service is actually from the intended app. i.e. It guarantees that a backend service knows that it's only servicing requests from the associated mobile app."]
    }), "\n", jsxs(_components.p, {
      children: ["At a high level, this is done by generating a new signing keypair on the device's secure enclave ", jsx(_components.em, {
        children: "that only the specific mobile app can access"
      }), ". There is then a protocol for Apple to ", jsx(_components.em, {
        children: "attest"
      }), " to the keypair as having been generated on said mobile app. This ensures any subsequent signatures using the keypair is guaranteed to be from the app."]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/diagram_appattest.png",
        alt: "App Attest protocol diagram"
      })
    }), "\n", jsx(_components.p, {
      children: "Specifically, the App Attest protocol provides an attestation from Apple's backend server that is bound to some unique nonce and the public key of the secure enclave keypair. The app can then send this attestation as part of a request, such that the receiving backend service can verify the attestation."
    }), "\n", jsx(_components.p, {
      children: "Note, ZCAM leverages Apple App Attest, but uses it in a subtle but different way. There are two key differences:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The unique nonce used in the attestation is actually derived on the app itself, instead of receiving it from some backend service."
      }), "\n", jsx(_components.li, {
        children: "The attestation can be verified either directly (bindings verification) or in a zero knowledge proof (ZK proof verification) instead of by a backend service."
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "ios-secure-enclave",
      children: ["iOS Secure Enclave", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ios-secure-enclave",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The Secure Enclave is a coprocessor found in Apple devices that provides an isolated, secure environment for cryptographic operations. Keys stored in the Secure Enclave:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Cannot be extracted or copied"
      }), "\n", jsx(_components.li, {
        children: "Are protected even if the device is compromised"
      }), "\n", jsx(_components.li, {
        children: "Can only be used by the specific app that created them"
      }), "\n", jsx(_components.li, {
        children: "Are hardware-backed and tamper-resistant"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "ZCAM uses the Secure Enclave to store both the device key (used for signing photo hashes) and the content key (used for signing C2PA manifests). This ensures that even if an attacker gains access to the device, they cannot forge signatures or create fake zphotos."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
