// vocs/_lib/vite/.vocs/dist/assets/how_it_works-BqDGR-01.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "How it Works",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "how-it-works",
        children: ["How it Works", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#how-it-works",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "ZCAM verifies photo authenticity via a combination of hardware-backed attestation, cryptographic signatures, and optional zero-knowledge proofs."
    }), "\n", jsx(_components.p, {
      children: "When a user takes a photo, ZCAM signs it using an Apple attested, secure enclave key. This attestation is embedded into the photo using the C2PA standard. Optionally, the attestation can be wrapped in a zero-knowledge proof for enhanced privacy."
    }), "\n", jsx(_components.p, {
      children: "The system works in three steps:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Capture"
        }), ": Take a photo and sign it with a hardware-backed key, attaching the attestation to the photo."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Prove"
        }), " (optional): Generate a zero-knowledge proof of the signature"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Verify"
        }), ": Validate the photo's authenticity using either bindings or ZK proof verification"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "verification-modes",
      children: ["Verification Modes", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verification-modes",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Bindings verification"
        }), " verifies the Apple App Attest signature directly. It's faster, works offline, and requires no zero-knowledge proofs."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "ZK proof verification"
        }), " generates a zero-knowledge proof for enhanced privacy, but requires network access for proof generation."]
      }), "\n"]
    }), "\n", jsx(_components.aside, {
      "data-callout": "info",
      children: jsxs(_components.p, {
        children: ["ZK proofs are ", jsx(_components.strong, {
          children: "optional"
        }), ": you can use ZCAM with bindings verification alone, or add ZK proofs when you need additional privacy guarantees."]
      })
    }), "\n", jsxs(_components.h2, {
      id: "components",
      children: ["Components", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#components",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM is built on three core technologies: Apple's App Attest service and Secure Enclave for hardware-backed signing, C2PA for embedding metadata, and SP1 for zero-knowledge proofs."
    }), "\n", jsxs(_components.h3, {
      id: "apple-app-attest-and-secure-enclave",
      children: ["Apple App Attest and Secure Enclave", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#apple-app-attest-and-secure-enclave",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM uses the Secure Enclave to store signing keys and Apple's App Attest service to ensure that a photo was signed by the ZCAM SDK."
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "App Attest"
      }), " is designed to let backend services verify that requests are coming from a legitimate app. It works by generating a signing keypair in the device's secure enclave that only the specific app can access. Apple then attests that the keypair was generated on that app and that the app itself has not been tamepred such that any subsequent signatures are guaranteed to originate from it."]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/images/diagram_appattest.png",
        alt: "App Attest protocol diagram"
      })
    }), "\n", jsx(_components.p, {
      children: "The App Attest protocol provides an attestation from Apple's servers, bound to a unique nonce and the public key of the secure enclave keypair. The app attaches this attestation to requests so the receiving service can verify authenticity."
    }), "\n", jsx(_components.p, {
      children: "ZCAM uses App Attest differently in two ways:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["The nonce is unique ", jsx(_components.em, {
          children: "per photo"
        }), " and is derived on the device itself, rather than received from a backend service."]
      }), "\n", jsx(_components.li, {
        children: "The attestation can be verified directly (bindings verification) or inside a zero-knowledge proof (ZK verification), rather than by a backend service."
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["The ", jsx(_components.strong, {
        children: "Secure Enclave"
      }), " is a coprocessor in Apple devices that provides an isolated environment for cryptographic operations. Keys stored in the Secure Enclave cannot be extracted or copied, are protected even if the device is compromised, and can only be used by the app that created them.\nZCAM stores both the device key (for signing photo hashes) and the content key (for signing C2PA manifests) in the Secure Enclave. This ensures that even if an attacker gains device access, they cannot forge signatures using the ZCAM signing key."]
    }), "\n", jsxs(_components.h3, {
      id: "c2pa",
      children: ["C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "C2PA is an open standard for attaching metadata that is cryptographically bound to an image. A C2PA manifest contains details about the photo, such as device, dimensions, history, and a cryptographic signature over those details."
    }), "\n", jsx(_components.p, {
      children: "The manifest is either embedded directly into the photo file in some metadata section of the file or can be stored separately and referenced using an invisible watermark."
    }), "\n", jsx(_components.p, {
      children: "ZCAM uses C2PA to embed attestation data (and optionally zero-knowledge proofs), proving that the photo was taken on an iPhone and signed using a secure enclave key that only the ZCAM SDK can access."
    }), "\n", jsxs(_components.h3, {
      id: "sp1-zero-knowledge-proofs",
      children: ["SP1 Zero-Knowledge Proofs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#sp1-zero-knowledge-proofs",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Zero-knowledge proofs let one party prove to another that a statement is true without revealing anything beyond the statement's validity."
    }), "\n", jsxs(_components.p, {
      children: ["ZCAM uses ", jsx(_components.a, {
        href: "https://docs.succinct.xyz/docs/sp1/introduction",
        children: "SP1"
      }), ", Succinct's zero-knowledge virtual machine (zkVM). SP1 proves the correct execution of programs compiled for the RISC-V architecture, meaning it can run and prove programs written in Rust, C++, C, or any language that compiles to RISC-V."]
    }), "\n", jsx(_components.p, {
      children: "SP1 offers three key benefits:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Maintainability"
        }), ": Write ZK programs in standard Rust without custom DSLs or complex circuits"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Faster Development"
        }), ": Skip months of low-level ZK engineering"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Performance"
        }), ": State-of-the-art proving speed and efficiency, benchmarked in production"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "In ZCAM, SP1 generates proofs that verify the photo was signed with a valid Apple App Attest key, the signature matches the photo hash, and the attestation chain is valid. These proofs are small, fast to verify, and don't reveal the underlying attestation data."
    }), "\n", jsxs(_components.p, {
      children: ["As mentioned above, zero knowledge proofs are ", jsx(_components.em, {
        children: "optional"
      }), ". You can use ZCAM with bindings verification alone. ZK proofs add privacy guarantees but require internet access to request proofs from the Succinct Prover Network. There is also higher latency given the time required for the network to prove the verification of an attestation."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
