// vocs/_lib/vite/.vocs/dist/assets/image-metadata-stripping-D1h9mDhf.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Image Metadata Stripping",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    header: "header",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "image-metadata-stripping",
        children: ["Image Metadata Stripping", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#image-metadata-stripping",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "This table documents how different platforms handle C2PA content credentials when images are uploaded and accessed through various methods."
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Platform"
            })
          }), jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Upload Method"
            })
          }), jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Read Method"
            })
          }), jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Maintains Credential?"
            })
          }), jsx(_components.th, {
            children: jsx(_components.strong, {
              children: "Notes"
            })
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Notion"
            })
          }), jsxs(_components.td, {
            children: ["Upload using ", jsx(_components.code, {
              children: "/file"
            })]
          }), jsxs(_components.td, {
            children: ["Click on link, hosted on ", jsx(_components.code, {
              children: "https://file.notion.so/..."
            })]
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Notion"
            })
          }), jsxs(_components.td, {
            children: ["Upload using ", jsx(_components.code, {
              children: "/file"
            })]
          }), jsx(_components.td, {
            children: "Download"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Twitter"
            })
          }), jsx(_components.td, {
            children: "Upload using upload file functionality"
          }), jsx(_components.td, {
            children: "Download"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Twitter"
            })
          }), jsx(_components.td, {
            children: "Link to image"
          }), jsx(_components.td, {
            children: "Click on Link"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Doesn't render properly unless hosting has og tag"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Instagram"
            })
          }), jsx(_components.td, {
            children: "New post (ensure original ratio/ not cropped etc.)"
          }), jsx(_components.td, {
            children: "No way to download, can expect src to get uploaded image"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "LinkedIn"
            })
          }), jsx(_components.td, {
            children: "Upload in post"
          }), jsx(_components.td, {
            children: "Natively embedded"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "LinkedIn"
            })
          }), jsx(_components.td, {
            children: "Upload in post"
          }), jsx(_components.td, {
            children: "Download image"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Discord"
            })
          }), jsx(_components.td, {
            children: "Upload file"
          }), jsx(_components.td, {
            children: "Open hosted link"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "Strips metadata"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Discord"
            })
          }), jsx(_components.td, {
            children: "Upload file"
          }), jsx(_components.td, {
            children: "Download file from hosted link"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Gmail"
            })
          }), jsx(_components.td, {
            children: "Attach file to email"
          }), jsx(_components.td, {
            children: "Click preview"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Gmail"
            })
          }), jsx(_components.td, {
            children: "Attach file to email"
          }), jsx(_components.td, {
            children: "Download"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Gmail"
            })
          }), jsx(_components.td, {
            children: "Embed file in email"
          }), jsx(_components.td, {
            children: "Download"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {})]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Gmail"
            })
          }), jsx(_components.td, {
            children: "Embed file in email"
          }), jsx(_components.td, {
            children: "Click preview"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {})]
        })]
      })]
    }), "\n", jsx(_components.p, {
      children: "Note: Cloudflare Images also supports C2PA."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
