// vocs/_lib/vite/.vocs/dist/assets/index-DAi4CwiS.js
import { jsx, jsxs } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
import "react";
import { L as Link, b as button$1, a as button_accent, u as useConfig, c as Logo$1, R as Root$1, d as List, T as Trigger, C as Content } from "../index.server.js";
import clsx from "clsx";
import "nuqs/adapters/react";
import "react-router";
import "node:fs";
import "node:path";
import "toml";
import "vite";
import "@vanilla-extract/dynamic";
import "react-intersection-observer";
import "@mdx-js/mdx";
import "@radix-ui/react-icons";
import "@radix-ui/react-dialog";
import "nuqs";
import "minisearch";
import "@radix-ui/react-label";
import "mark.js";
import "@radix-ui/react-navigation-menu";
import "@radix-ui/react-accordion";
import "radix-ui";
import "@radix-ui/react-popover";
import "@mdx-js/react";
import "@radix-ui/react-tabs";
import "@floating-ui/react";
import "picomatch";
function Button$1({ children, className, href, variant }) {
  return jsx(Link, { className: clsx(className, button$1, variant === "accent" && button_accent), href, variant: "styleless", children });
}
var root = "vocs_HomePage";
var logo = "vocs_HomePage_logo";
var title = "vocs_HomePage_title";
var tagline = "vocs_HomePage_tagline";
var description = "vocs_HomePage_description";
var buttons = "vocs_HomePage_buttons";
var button = "vocs_HomePage_button";
var tabs = "vocs_HomePage_tabs";
var tabsList = "vocs_HomePage_tabsList";
var tabsContent = "vocs_HomePage_tabsContent";
var packageManager = "vocs_HomePage_packageManager";
function Root({ children, className }) {
  return jsx("div", { className: clsx(className, root), children });
}
function Logo({ className }) {
  const { logoUrl, title: title$1 } = useConfig();
  return logoUrl ? jsx("div", { className: clsx(className, logo), children: jsx(Logo$1, {}) }) : jsx("h1", { className: clsx(className, title), children: title$1 });
}
function Tagline({ children, className }) {
  return jsx("div", { className: clsx(className, tagline), children });
}
function Description({ children, className }) {
  return jsx("div", { className: clsx(className, description), children });
}
function Buttons({ children, className }) {
  return jsx("div", { className: clsx(className, buttons), children });
}
function Button(props) {
  return jsx(Button$1, { ...props, className: clsx(button, props.className) });
}
function InstallPackage({ name, type = "install" }) {
  return jsxs(Root$1, { className: tabs, defaultValue: "npm", children: [jsxs(List, { className: tabsList, children: [jsx(Trigger, { value: "npm", children: "npm" }), jsx(Trigger, { value: "pnpm", children: "pnpm" }), jsx(Trigger, { value: "yarn", children: "yarn" })] }), jsxs(Content, { className: tabsContent, value: "npm", children: [jsx("span", { className: packageManager, children: "npm" }), " ", type === "init" ? "init" : "install", " ", name] }), jsxs(Content, { className: tabsContent, value: "pnpm", children: [jsx("span", { className: packageManager, children: "pnpm" }), " ", type === "init" ? "create" : "add", " ", name] }), jsxs(Content, { className: tabsContent, value: "yarn", children: [jsx("span", { className: packageManager, children: "yarn" }), " ", type === "init" ? "create" : "add", " ", name] })] });
}
const HomePage = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  Button,
  Buttons,
  Description,
  InstallPackage,
  Logo,
  Root,
  Tagline
}, Symbol.toStringTag, { value: "Module" }));
const frontmatter = {
  "layout": "landing"
};
function _createMdxContent(props) {
  if (!HomePage) _missingMdxReference("HomePage", false);
  if (!Button) _missingMdxReference("HomePage.Button", true);
  if (!Buttons) _missingMdxReference("HomePage.Buttons", true);
  if (!Description) _missingMdxReference("HomePage.Description", true);
  if (!Logo) _missingMdxReference("HomePage.Logo", true);
  if (!Root) _missingMdxReference("HomePage.Root", true);
  if (!Tagline) _missingMdxReference("HomePage.Tagline", true);
  return jsxs(Root, {
    children: [jsx(Logo, {}), jsx(Tagline, {
      children: "ZCAM SDK Documentation"
    }), jsx(Description, {
      children: "Take and verify authentic photos with zero-knowledge proofs"
    }), jsxs(Buttons, {
      children: [jsx(Button, {
        href: "/overview/introduction",
        variant: "accent",
        children: "Get started"
      }), jsx(Button, {
        href: "https://github.com/succinctlabs/zcam1-sdk",
        children: "GitHub"
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent();
}
function _missingMdxReference(id, component) {
  throw new Error("Expected " + (component ? "component" : "object") + " `" + id + "` to be defined: you likely forgot to import, pass, or provide it.");
}
export {
  MDXContent as default,
  frontmatter
};
