// vocs/_lib/vite/.vocs/dist/assets/installation-jP6ACpEe.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Installation",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    pre: "pre",
    span: "span",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "installation",
        children: ["Installation", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#installation",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "prerequisites",
      children: ["Prerequisites", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#prerequisites",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "rust-toolchain",
      children: ["Rust Toolchain", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#rust-toolchain",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM SDK packages include Rust bindings built via UniFFI. You need Rust installed with iOS targets:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Install Rust (if not already installed)"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "curl"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --proto"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " '=https'"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --tlsv1.2"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " -sSf"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " https://sh.rustup.rs"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: " |"
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: " sh"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Add iOS targets"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "rustup"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " target"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " add"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " aarch64-apple-ios"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " aarch64-apple-ios-sim"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "xcode",
      children: ["Xcode", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#xcode",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Xcode is required for iOS development. Install it from the Mac App Store."
    }), "\n", jsxs(_components.h3, {
      id: "development-environment",
      children: ["Development Environment", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#development-environment",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Node.js 18+"
      }), "\n", jsx(_components.li, {
        children: "React Native 0.81+"
      }), "\n", jsx(_components.li, {
        children: "Expo SDK 54+ (if using Expo)"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "installing-packages",
      children: ["Installing Packages", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#installing-packages",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM SDK is divided into standalone packages. Install only what you need:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Full flow (capture, prove, verify)"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "npm"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " i"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " react-native-zcam1-capture"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " react-native-zcam1-prove"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " react-native-zcam1-verify"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Install native dependencies"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: "cd"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " ios"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: " && "
            }), jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "pod"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " install"
            })]
          })]
        })
      })
    }), "\n", jsx(_components.p, {
      children: "Individual packages:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Package"
          }), jsx(_components.th, {
            children: "Purpose"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.code, {
              children: "react-native-zcam1-capture"
            })
          }), jsx(_components.td, {
            children: "Take photos with C2PA manifests"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.code, {
              children: "react-native-zcam1-prove"
            })
          }), jsx(_components.td, {
            children: "Generate ZK proofs for photos"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.code, {
              children: "react-native-zcam1-verify"
            })
          }), jsx(_components.td, {
            children: "Verify photo authenticity"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.code, {
              children: "react-native-zcam1-picker"
            })
          }), jsx(_components.td, {
            children: "Image picker with authenticity badges"
          })]
        })]
      })]
    }), "\n", jsx(_components.aside, {
      "data-callout": "warning",
      children: jsxs(_components.p, {
        children: ["The UniFFI bindings are built during ", jsx(_components.code, {
          children: "npm install"
        }), ". This can take several minutes on first install."]
      })
    }), "\n", jsxs(_components.h2, {
      id: "ios-configuration",
      children: ["iOS Configuration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ios-configuration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "bundle-identifier",
      children: ["Bundle Identifier", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#bundle-identifier",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Configure your app's bundle identifier in your project. For Expo apps, set it in ", jsx(_components.code, {
        children: "app.json"
      }), ":"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "expo"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '    "ios"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "bundleIdentifier"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"com.yourcompany.yourapp"'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h3, {
      id: "app-id-for-app-attest",
      children: ["App ID for App Attest", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#app-id-for-app-attest",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "You'll need your Apple Team ID combined with your bundle identifier. The format is:"
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsx(_components.code, {
          children: jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "TEAM_ID"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">.<"
            }), jsx(_components.span, {
              style: {
                color: "#22863A",
                "--shiki-dark": "#8DDB8C"
              },
              children: "BUNDLE_IDENTIFIER"
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ">"
            })]
          })
        })
      })
    }), "\n", jsxs(_components.p, {
      children: ["For example: ", jsx(_components.code, {
        children: "NLS5R4YCGX.com.yourcompany.yourapp"
      })]
    }), "\n", jsx(_components.p, {
      children: "To find your Team ID:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Open Keychain Access on your Mac"
      }), "\n", jsx(_components.li, {
        children: 'Find a certificate named "Apple Development"'
      }), "\n", jsx(_components.li, {
        children: 'The Team ID is the "Organizational Unit" field'
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "required-permissions",
      children: ["Required Permissions", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#required-permissions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Add these keys to your ", jsx(_components.code, {
        children: "Info.plist"
      }), " (or ", jsx(_components.code, {
        children: "app.json"
      }), " for Expo):"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "{"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '  "expo"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '    "ios"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '      "infoPlist"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": {"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '        "NSCameraUsageDescription"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"This app uses your camera to capture photos."'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ","
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#8DDB8C"
              },
              children: '        "NSPhotoLibraryUsageDescription"'
            }), jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: ": "
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: '"This app needs access to save photos."'
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "      }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "    }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "  }"
            })
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "}"
            })
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "environment-variables",
      children: ["Environment Variables", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#environment-variables",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Create a ", jsx(_components.code, {
        children: ".env"
      }), " file with your configuration:"]
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "EXPO_PUBLIC_APP_ID"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "=<"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "TEAM_ID"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ">"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "."
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "<"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "BUNDLE_IDENTIFIER"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: ">"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#24292E",
                "--shiki-dark": "#ADBAC7"
              },
              children: "EXPO_PUBLIC_BACKEND_URL"
            }), jsx(_components.span, {
              style: {
                color: "#D73A49",
                "--shiki-dark": "#F47067"
              },
              children: "="
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: "https://your-backend-url.com"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.h2, {
      id: "running-on-device",
      children: ["Running on Device", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#running-on-device",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "App Attest requires a physical iOS device—it does not work in the simulator (the SDK falls back to mock mode for development)."
    }), "\n", jsx(Fragment, {
      children: jsx(_components.pre, {
        className: "shiki shiki-themes github-light github-dark-dimmed",
        style: {
          backgroundColor: "#fff",
          "--shiki-dark-bg": "#22272e",
          color: "#24292e",
          "--shiki-dark": "#adbac7"
        },
        tabIndex: "0",
        children: jsxs(_components.code, {
          children: [jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# Expo"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "npx"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " expo"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " prebuild"
            })]
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "npx"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " expo"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " run:ios"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --device"
            })]
          }), "\n", jsx(_components.span, {
            className: "line",
            "data-empty-line": true,
            children: " "
          }), "\n", jsx(_components.span, {
            className: "line",
            children: jsx(_components.span, {
              style: {
                color: "#6A737D",
                "--shiki-dark": "#768390"
              },
              children: "# React Native CLI"
            })
          }), "\n", jsxs(_components.span, {
            className: "line",
            children: [jsx(_components.span, {
              style: {
                color: "#6F42C1",
                "--shiki-dark": "#F69D50"
              },
              children: "npx"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " react-native"
            }), jsx(_components.span, {
              style: {
                color: "#032F62",
                "--shiki-dark": "#96D0FF"
              },
              children: " run-ios"
            }), jsx(_components.span, {
              style: {
                color: "#005CC5",
                "--shiki-dark": "#6CB6FF"
              },
              children: " --device"
            })]
          })]
        })
      })
    }), "\n", jsxs(_components.p, {
      children: ["On first run, you may need to trust the developer certificate on your device:\n", jsx(_components.strong, {
        children: "Settings → General → Device Management → Trust"
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
