// vocs/_lib/vite/.vocs/dist/assets/introduction-CVSTYKsg.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Introduction",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    li: "li",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "introduction",
        children: ["Introduction", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#introduction",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "In an era of AI-generated images and sophisticated editing tools, proving that a photo is real matters more than ever. ZCAM gives developers the tools to build authentic images directly into their apps."
    }), "\n", jsx(_components.p, {
      children: "The SDK is built on three core technologies:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Apple's App Attest"
        }), " guarantees that signing keys are generated and stored in the iPhone's secure enclave, tied to a specific app."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "C2PA"
        }), " embeds verifiable provenance metadata, including capture details and edit history, directly into the image file."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "SP1 zero-knowledge proofs"
        }), " optionally wrap verification into succinct, easily verifiable proofs using Succinct's ", jsx(_components.a, {
          href: "https://docs.succinct.xyz/docs/sp1/introduction",
          children: "SP1"
        }), " zkVM."]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Whether you're building for journalism, insurance, identity verification, or any use case where image authenticity matters, these docs will help you get started."
    }), "\n", jsx("style", {
      children: `
.doc-cards {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  gap: 1.5rem;
  margin-top: 2rem;
  margin-bottom: 2rem;
}
.doc-card {
  display: block;
  padding: 1.5rem;
  border: 1px solid var(--vocs-color_border);
  border-radius: 8px;
  text-decoration: none;
  color: inherit;
  transition: all 0.2s ease;
  background-color: var(--vocs-color_background);
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
}
.doc-card:hover {
  border-color: var(--vocs-color_accent);
  transform: translateY(-2px);
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
}
.doc-card h3 {
  margin-top: 0;
  margin-bottom: 0.5rem;
  color: var(--vocs-color_accent);
}
.doc-card p {
  margin: 0;
  font-size: 0.9rem;
  opacity: 0.8;
}
`
    }), "\n", jsxs("div", {
      className: "doc-cards",
      children: [jsxs("a", {
        href: "/getting-started/installation",
        className: "doc-card",
        children: [jsx("h3", {
          children: "🚀 Getting Started"
        }), jsx("p", {
          children: jsx(_components.p, {
            children: "Install the SDK and get up and running with your first zphoto in minutes."
          })
        })]
      }), jsxs("a", {
        href: "/sdk/capture",
        className: "doc-card",
        children: [jsx("h3", {
          children: "📱 Using the SDK"
        }), jsx("p", {
          children: jsx(_components.p, {
            children: "Learn how to build apps with the ZCAM SDK."
          })
        })]
      }), jsxs("a", {
        href: "/c2pa-ecosystem/editing_zcam_photos",
        className: "doc-card",
        children: [jsx("h3", {
          children: "🔗 C2PA Compatibility"
        }), jsx("p", {
          children: jsx(_components.p, {
            children: "Understand how ZCAM works with C2PA and editing tools in the ecosystem."
          })
        })]
      }), jsxs("a", {
        href: "/technical-docs/architecture",
        className: "doc-card",
        children: [jsx("h3", {
          children: "🔧 Technical Docs"
        }), jsx("p", {
          children: jsx(_components.p, {
            children: "Deep dive into the ZCAM architecture, security, and FAQs."
          })
        })]
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
