// vocs/_lib/vite/.vocs/dist/assets/introduction-DcATMuSn.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Introduction",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    header: "header",
    li: "li",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "introduction",
        children: ["Introduction", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#introduction",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Today, with the advent of AI generated images, liberal usage of editing software and increasing incentives to deceive and alter public perception, the importance of image integrity is at an all time high."
    }), "\n", jsx(_components.p, {
      children: "Succinct's ZCAM allows users to use their iPhones to create zphotos: These photos can be verified as authentic, unedited and indeed originating from their iPhone's camera. The ZCAM SDK allows any app to integrate this functionality, enabling them to leverage the ability take and verify zphotos in their app."
    }), "\n", jsx(_components.p, {
      children: "The core technology of zphotos and ZCAM is built on three components:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "SP1 Zero Knowledge proofs"
        }), ": Succinct's zero knowledge virtual machine can generate proofs that verify that certain code was ran as intended, and outputs succinct, small easily verifiable proofs"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Apple App Attest"
        }), ": The App Attest service allows an app to guarantee that a signing keypair is generated and stored on an Iphone's secure enclave, and belongs to a ", jsx(_components.em, {
          children: "specific"
        }), " app on the iPhone."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "C2PA"
        }), ": C2PA is a standard that specifies how to embed verifiable metadata about an image within the actual image file. This includes details about the image, the history of the image etc."]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM sdk allows users to take a photo, sign the photo using an Apple attested key, and then embeds a zero knowledge proof verifying the signature along with the validity of the attested key to the image using C2PA."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
