// vocs/_lib/vite/.vocs/dist/assets/landscape-Dfa_grZw.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Landscape",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "landscape",
        children: ["Landscape", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#landscape",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Several technologies exist for image authenticity and provenance. Most solve for different problems, such as edit history, AI detection, trust scoring, rather than proving a photo was authentically captured on a real device."
    }), "\n", jsx(_components.p, {
      children: "ZCAM is a complete solution for mobile photo authenticity. It combines hardware-backed attestation, optional zero-knowledge proofs, and embedded verification into a single system designed for human-taken photos on mobile devices."
    }), "\n", jsxs(_components.h2, {
      id: "comparison",
      children: ["Comparison", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#comparison",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Approach"
          }), jsx(_components.th, {
            children: "Hardware Attestation"
          }), jsx(_components.th, {
            children: "ZK Proofs"
          }), jsx(_components.th, {
            children: "Embedded Verification"
          }), jsx(_components.th, {
            children: "User-friendly Output"
          }), jsx(_components.th, {
            children: "Camera Photos"
          }), jsx(_components.th, {
            children: "Use Case"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "ZCAM"
            })
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Complete mobile photo authenticity solution"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "C2PA"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Edit history and provenance"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Physical Cameras"
            })
          }), jsx(_components.td, {
            children: "❓"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Hardware signed capture with limited support"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "JPEG Trust"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "Trust evaluation framework"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "SynthID"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "AI watermarking and detection"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "AI Content Marking"
            })
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "✅"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "❌"
          }), jsx(_components.td, {
            children: "Labeling AI-generated content"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "other-approaches",
      children: ["Other Approaches", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#other-approaches",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "c2pa",
      children: ["C2PA", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#c2pa",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "C2PA is a standard for creating a verifiable history of an image. It is not specific to authentic photos and applies equally to camera captures and AI-generated images. As an image evolves (cropping, brightness adjustments, etc.), C2PA-compatible editors attach new signed manifests tracking each change."
    }), "\n", jsx(_components.p, {
      children: "ZCAM uses C2PA as its container format but adds hardware-backed guarantees. Where C2PA tracks history with any signing key, ZCAM guarantees the photo was captured on a specific device using keys that can't be extracted or forged."
    }), "\n", jsxs(_components.h3, {
      id: "jpeg-trust",
      children: ["JPEG Trust", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#jpeg-trust",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "JPEG Trust is a framework for evaluating image trustworthiness. It works on top of C2PA, using manifest data to generate trust scores. It is an evaluation layer, not a capture or verification system, and could work alongside ZCAM-generated manifests."
    }), "\n", jsxs(_components.h3, {
      id: "synthid",
      children: ["SynthID", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#synthid",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "SynthID, developed by Google, embeds invisible watermarks into AI-generated images. Unlike metadata-based approaches, these watermarks survive cropping and editing. A separate model analyzes images to determine whether they're AI-generated.\nSynthID focuses on AI detection with probabilistic results. ZCAM focuses on authenticity with cryptographic guarantees."
    }), "\n", jsxs(_components.h3, {
      id: "ai-content-marking",
      children: ["AI Content Marking", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ai-content-marking",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Several companies provide solutions for marking AI-generated content, using different approaches:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Meta"
        }), " adds durable watermarks specifically to AI-generated content."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "OpenAI"
        }), " generates C2PA-compatible images with DALL-E and ChatGPT."]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "These solutions label AI content, whereas ZCAM verifies human-captured photos."
    }), "\n", jsxs(_components.h2, {
      id: "when-to-use-zcam",
      children: ["When to Use ZCAM", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#when-to-use-zcam",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "ZCAM is the right choice when you need:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Hardware-backed proof that a photo was taken on a real device"
      }), "\n", jsx(_components.li, {
        children: "Cryptographic verification rather than probabilistic detection"
      }), "\n", jsx(_components.li, {
        children: "A mobile-first solution"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Other approaches may be more optimized for marking AI-generated content (SynthID, AI Content Marking), tracking edit history (C2PA), or non-mobile platforms."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
