// vocs/_lib/vite/.vocs/dist/assets/security-OL0eSNJQ.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Security",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    div: "div",
    em: "em",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "security",
        children: ["Security", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#security",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "security-claims",
      children: ["Security Claims", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#security-claims",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "A verified zphoto provides the following guarantees:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Device Authenticity"
        }), ": The photo was signed on a genuine Apple device with a Secure Enclave"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "App Binding"
        }), ": The signing key is bound to a specific application via Apple App Attest"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Content Integrity"
        }), ": The photo has not been modified since the signature was created"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Proof Validity"
        }), ": The above properties are verified in a zero-knowledge proof that chains to Apple's root certificate"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "trust-assumptions",
      children: ["Trust Assumptions", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#trust-assumptions",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The security of ZCAM relies on the following assumptions:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Assumption"
          }), jsx(_components.th, {
            children: "Rationale"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "Secure Enclave keys cannot be exfiltrated"
          }), jsx(_components.td, {
            children: "Hardware security guarantee from Apple"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "Apple App Attest is honest"
          }), jsx(_components.td, {
            children: "Apple signs attestations only for legitimate device/app pairs"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "SP1 proof system is sound"
          }), jsx(_components.td, {
            children: "Groth16 proofs cannot be forged without the witness"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "SDK code integrity"
          }), jsx(_components.td, {
            children: "The SDK code on-device has not been tampered with"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "critical-assumption-capture-to-sign-integrity",
      children: ["Critical Assumption: Capture-to-Sign Integrity", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#critical-assumption-capture-to-sign-integrity",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.aside, {
      "data-callout": "warning",
      children: jsx(_components.p, {
        children: "The SDK signs the photo immediately after capture, but there is no cryptographic binding\nbetween the camera hardware and the signing operation. We assume no malicious code can\nexecute between the moment the photo is captured and when it is signed."
      })
    }), "\n", jsx(_components.p, {
      children: "This assumption holds if:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The SDK code has not been tampered with"
      }), "\n", jsx(_components.li, {
        children: "The host application does not inject code into the capture flow"
      }), "\n", jsx(_components.li, {
        children: "The device is not compromised"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: ["This is an ", jsx(_components.strong, {
        children: "operational security"
      }), " assumption, not a cryptographic guarantee."]
    }), "\n", jsxs(_components.h2, {
      id: "open-issues",
      children: ["Open Issues", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#open-issues",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "app-trust-model",
      children: ["App Trust Model", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#app-trust-model",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Currently, the SP1 program verifies that a photo was signed by ", jsx(_components.em, {
        children: "an"
      }), " Apple Attested key,\nbut does not restrict ", jsx(_components.em, {
        children: "which"
      }), " apps are trusted. A malicious app could:"]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "Integrate the SDK"
      }), "\n", jsx(_components.li, {
        children: "Feed AI-generated images to the signing flow"
      }), "\n", jsx(_components.li, {
        children: "Generate valid proofs for inauthentic photos"
      }), "\n"]
    }), "\n", jsx(_components.strong, {
      children: "Mitigations under consideration:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Maintain a allowlist of trusted App IDs verified in the proof"
      }), "\n", jsx(_components.li, {
        children: "Output the App ID as a public value, allowing verifiers to check trust"
      }), "\n", jsx(_components.li, {
        children: "Require app attestation/audit before allowlist inclusion"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "third-party-sdk-integration",
      children: ["Third-Party SDK Integration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#third-party-sdk-integration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "As the SDK scales to third-party apps, we need mechanisms to ensure integrators\nuse the SDK as intended. Options include:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Code signing / integrity checks on the SDK binary"
      }), "\n", jsx(_components.li, {
        children: "Runtime attestation of the SDK version"
      }), "\n", jsx(_components.li, {
        children: "Legal/contractual requirements for integrators"
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "sdk-code-integrity",
      children: ["SDK Code Integrity", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#sdk-code-integrity",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "We need to guarantee the SDK code cannot be tampered with:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Build pipeline"
        }), ": CI/CD must produce reproducible, auditable builds"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "On-device"
        }), ": The SDK binary should be verified before execution"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Updates"
        }), ": SDK updates must be authenticated"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "This is not yet fully implemented."
    }), "\n", jsxs(_components.h2, {
      id: "inauthentic-photos",
      children: ["Inauthentic Photos", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#inauthentic-photos",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Even with cryptographic guarantees that a photo was captured by a genuine device running the SDK, attackers may attempt physical attacks:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Screen capture"
        }), ": Photographing a screen displaying an AI-generated or manipulated image"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Printed photo capture"
        }), ": Photographing a printed image"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Projector capture"
        }), ": Photographing a projected image"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: 'These attacks bypass software-level protections because the camera genuinely captures "something" — just not an authentic scene.'
    }), "\n", jsx(_components.p, {
      children: 'Note that the taken photo is still "valid" in that it was taken using the camera app, properly signed etc. But the photo is still "inauthentic" in terms of what it is trying to portray.\nThis is a different, much harder problem to solve. In fact, a cryptographic solution is likely impossible. Instead, our strategy is to provide as much contextual data in the metadata for a user to be able to make an informed decision on the authenticity of the photo.'
    }), "\n", jsxs(_components.h3, {
      id: "detection-signals",
      children: ["Detection Signals", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#detection-signals",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The following metadata fields can help detect physical replay attacks:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Field"
          }), jsx(_components.th, {
            children: "What It Measures"
          }), jsx(_components.th, {
            children: "Fraud Indicator"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "GPS / Location"
            })
          }), jsx(_components.td, {
            children: "Location"
          }), jsx(_components.td, {
            children: "Location inconsistent with claimed scene"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Timestamp"
            })
          }), jsx(_components.td, {
            children: "Time"
          }), jsx(_components.td, {
            children: "Time inconsistent with lighting conditions"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Subject Distance"
            })
          }), jsx(_components.td, {
            children: "Distance to focused subject"
          }), jsx(_components.td, {
            children: "Focus distance inconsistent with the subject of picture"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Depth Map"
            })
          }), jsx(_components.td, {
            children: "Per-pixel depth from dual cameras/LiDAR"
          }), jsx(_components.td, {
            children: "Flat depth when the subject has more depth variance"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "White Balance"
            })
          }), jsx(_components.td, {
            children: "Color temperature"
          }), jsx(_components.td, {
            children: "Screen color temperature differs from natural light"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Ambient Light"
            })
          }), jsx(_components.td, {
            children: "Environmental light level"
          }), jsx(_components.td, {
            children: "Unusually uniform lighting could indicate screen glow"
          })]
        })]
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
