// vocs/_lib/vite/.vocs/dist/assets/use_cases-oAuqokue.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Use Cases",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    strong: "strong",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "use-cases",
        children: ["Use Cases", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#use-cases",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Below are two real-world examples of how the ZCAM SDK can be integrated to solve different verification challenges."
    }), "\n", jsxs(_components.p, {
      children: ["These examples highlight two distinct patterns: ", jsx(_components.strong, {
        children: "end-user verification"
      }), ", where readers or customers verify photos themselves, and ", jsx(_components.strong, {
        children: "internal verification"
      }), ", where verification happens in backend processing."]
    }), "\n", jsxs(_components.h2, {
      id: "verifiable-news-platform",
      children: ["Verifiable News Platform", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#verifiable-news-platform",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Edited and AI-generated photos are a growing concern for news consumers. A news platform can use ZCAM to let readers independently verify that photos are authentic and unedited."
    }), "\n", jsx(_components.p, {
      children: "The system is comprised of:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "A mobile app for field reporters to capture verifiable photos"
      }), "\n", jsx(_components.li, {
        children: "Integration with the platform's CMS to store and display photos"
      }), "\n", jsx(_components.li, {
        children: "Public-facing verification tools so readers can check authenticity themselves"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Why it matters"
      }), ": Readers can independently verify that photos are authentic, building trust in the platform's journalism."]
    }), "\n", jsxs(_components.h2, {
      id: "architecture",
      children: ["Architecture", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#architecture",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The news platform integrates ZCAM SDK in two places:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Mobile app"
        }), " A lightweight iOS app for reporters to use to capture verifiable photos"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Web platform"
        }), ": For readers to verify photo authenticity directly in articles"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "mobile-app",
      children: ["Mobile App", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mobile-app",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The news platform builds a specialized camera app for reporters using the ZCAM SDK:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Installation"
        }), ": Setup the SDK using the ", jsx(_components.a, {
          href: "/getting-started/installation",
          children: "installation guide"
        })]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Camera Setup"
        }), ": Use ", jsx(_components.code, {
          children: "react-native-zcam1-capture"
        }), " to initialize the camera:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "initCapture()"
            }), " to initialize device keys and attestation"]
          }), "\n", jsxs(_components.li, {
            children: ["Use the ", jsx(_components.code, {
              children: "<ZCamera>"
            }), " component to render the camera preview"]
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "takePhoto()"
            }), " to capture a photo with verifiable bindings"]
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Proof Generation"
        }), " (Optional): Use ", jsx(_components.code, {
          children: "react-native-zcam1-prove"
        }), " to generate zero-knowledge proofs:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Wrap your app with ", jsx(_components.code, {
              children: "ProverProvider"
            }), " and use the ", jsx(_components.code, {
              children: "useProver()"
            }), " hook"]
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "waitAndEmbedProof()"
            }), " to generate and embed the proof, or use ", jsx(_components.code, {
              children: "requestProof()"
            }), " + ", jsx(_components.code, {
              children: "useProofRequestStatus()"
            }), " for progress tracking"]
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Upload"
        }), ": Upload the zphoto to the news platform's backend for use in articles"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "web-platform",
      children: ["Web Platform", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#web-platform",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "On the news platform's website, articles display verifiable photos with verification capabilities:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Photo Display"
        }), ": Photos embedded in articles include C2PA content credentials", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Users with C2PA browser extensions can automatically see content credentials"
          }), "\n", jsx(_components.li, {
            children: "The platform can display verification status badges"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Client-Side Verification"
        }), ": Integrate ", jsx(_components.code, {
          children: "react-native-zcam1-verify"
        }), " (or web equivalent) to enable verification:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: 'Add a "Verify Photo" button for each photo in articles'
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "verifyHash()"
            }), " and ", jsx(_components.code, {
              children: "verifyProof()"
            }), " methods to validate authenticity"]
          }), "\n", jsx(_components.li, {
            children: 'Display verification results to readers (e.g., "✓ Verified Authentic")'
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Metadata Display"
        }), ": Show readers metadata from the C2PA manifest:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Capture timestamp, device information, app identifier"
          }), "\n", jsx(_components.li, {
            children: "Allow readers to inspect the full verification chain"
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "insurance-claim-verification",
      children: ["Insurance Claim Verification", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#insurance-claim-verification",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Insurance companies allow policyholders to file claims using photos from their mobile app. This has led to an increase in fraud where claimants manipulate photos before submission. ZCAM ensures that claim photos were taken on the policyholder's device using the official app, and haven't been edited since capture."
    }), "\n", jsx(_components.p, {
      children: "The system is comprised of:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Camera functionality within the insurance app using ZCAM for capture"
      }), "\n", jsx(_components.li, {
        children: "Backend verification that automatically validates all submitted photos"
      }), "\n", jsx(_components.li, {
        children: "Automated triage that flags claims with suspicious or unverifiable photos"
      }), "\n"]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Why it matters"
      }), ": Automatically detect and reject claims with manipulated photos, reducing fraud and manual review costs."]
    }), "\n", jsxs(_components.h3, {
      id: "architecture-1",
      children: ["Architecture", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#architecture-1",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The insurance app integrates ZCAM SDK in two places:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Mobile app"
        }), ": Policyholders capture zphotos during claim filing."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Backend processing"
        }), ": The insurance company verifies photos automatically during claim triage."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "mobile-app-1",
      children: ["Mobile App", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#mobile-app-1",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The ZCAM SDK integrates into the claim landing flow:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Installation"
        }), ": Follow the ", jsx(_components.a, {
          href: "/getting-started/installation",
          children: "Installation guide"
        }), " to set up the SDK"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Camera Integration"
        }), ": Inject the ", jsx(_components.code, {
          children: "<ZCamera>"
        }), " component into the claim flow where users must photograph their vehicle:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "The SDK provides camera preview functionality"
          }), "\n", jsxs(_components.li, {
            children: ["Call ", jsx(_components.code, {
              children: "takePhoto()"
            }), " to capture a verifiable photo"]
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Proof Generation"
        }), " (Optional): Use ", jsx(_components.code, {
          children: "react-native-zcam1-prove"
        }), " to attach zero-knowledge proofs:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Use ", jsx(_components.code, {
              children: "waitAndEmbedProof()"
            }), " to generate and embed proofs"]
          }), "\n", jsx(_components.li, {
            children: "This provides additional cryptographic guarantees and privacy"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Claim Submission"
        }), ": When the user completes the claim flow, attach verifiable photos to the claim filing and upload to the insurance company's backend"]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "backend-processing",
      children: ["Backend Processing", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#backend-processing",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "On the insurance company's backend, verification is integrated into the automated claim triage process:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Automated Verification"
        }), ": When a claim enters the triage flow, the system automatically verifies all attached photos:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: ["Integrate ", jsx(_components.code, {
              children: "react-native-zcam1-verify"
            }), " (or server-side equivalent) into the backend"]
          }), "\n", jsxs(_components.li, {
            children: ["For each photo, call ", jsx(_components.code, {
              children: "verifyHash()"
            }), " and ", jsx(_components.code, {
              children: "verifyProof()"
            }), " methods"]
          }), "\n", jsx(_components.li, {
            children: "Check that the photo was taken using the insurance company's app"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Metadata Analysis"
        }), ": Leverage metadata from the C2PA manifest:", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Verify location data matches the claim location"
          }), "\n", jsx(_components.li, {
            children: "Check timestamp aligns with the reported incident time"
          }), "\n", jsx(_components.li, {
            children: "Validate device and app information"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Claim Decision"
        }), ": Based on verification results:", "\n", jsxs(_components.ul, {
          children: ["\n", jsxs(_components.li, {
            children: [jsx(_components.strong, {
              children: "Verification passes"
            }), ": Proceed with normal claim processing"]
          }), "\n", jsxs(_components.li, {
            children: [jsx(_components.strong, {
              children: "Verification fails"
            }), ": Flag claim for manual review or deny if photos are inauthentic"]
          }), "\n"]
        }), "\n"]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
