// vocs/_lib/vite/.vocs/dist/assets/zkproof-details-BOggK8RE.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "ZK Proof Details",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    li: "li",
    ol: "ol",
    p: "p",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "zk-proof-details",
        children: ["ZK Proof Details", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#zk-proof-details",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "The SP1 program proves the following:"
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsx(_components.li, {
        children: "The photo bytes matches the expected bytes in the C2PA manifest"
      }), "\n", jsx(_components.li, {
        children: "The Apple Attest attestation is valid"
      }), "\n", jsx(_components.li, {
        children: "The assertion (i.e. signature of the photo hash using the attested key) is valid"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "The photo bytes are provided as input to the proof."
    }), "\n", jsxs(_components.h2, {
      id: "proof-logic",
      children: ["Proof Logic", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#proof-logic",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Step"
          }), jsx(_components.th, {
            children: "Operation"
          }), jsx(_components.th, {
            children: "Guarantees"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "1"
          }), jsx(_components.td, {
            children: "Extract manifest from photo bytes"
          }), jsx(_components.td, {
            children: "—"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "2"
          }), jsx(_components.td, {
            children: "Extract bindings and data hash from manifest"
          }), jsx(_components.td, {
            children: "—"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "3"
          }), jsx(_components.td, {
            children: "Compute photo hash"
          }), jsx(_components.td, {
            children: "—"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "4"
          }), jsx(_components.td, {
            children: "Check data hash == photo hash"
          }), jsx(_components.td, {
            children: "Manifest corresponds to this photo"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "5"
          }), jsx(_components.td, {
            children: "Validate attestation"
          }), jsx(_components.td, {
            children: "The attestation is valid and from Apple"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "6"
          }), jsx(_components.td, {
            children: "Validate assertion"
          }), jsx(_components.td, {
            children: "Photo hash was signed by the attested key"
          })]
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "attestation-validation",
      children: ["Attestation Validation", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#attestation-validation",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Guarantees:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The device key was generated in a genuine Apple Secure Enclave"
      }), "\n", jsxs(_components.li, {
        children: ["The key is bound to a specific app (via ", jsx(_components.code, {
          children: "app_id"
        }), " / RP ID)"]
      }), "\n", jsxs(_components.li, {
        children: ["The device is running a legitimate app (AAGUID = ", jsx(_components.code, {
          children: "appattest"
        }), " or ", jsx(_components.code, {
          children: "appattestdevelop"
        }), ")"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Validates:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Certificate chain verifies up to Apple Root CA"
      }), "\n", jsx(_components.li, {
        children: "Nonce matches the challenge"
      }), "\n", jsx(_components.li, {
        children: "Public key hash matches the key ID"
      }), "\n", jsx(_components.li, {
        children: "RP ID == SHA256(app_id)"
      }), "\n", jsx(_components.li, {
        children: "AAGUID is a valid Apple attestation identifier"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "assertion-validation",
      children: ["Assertion Validation", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#assertion-validation",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Guarantees:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "The photo corresponding to this hash was signed using the ZCAM SDK"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "Validates:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "ECDSA signature over the photo hash using the public key from attestation"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "public-outputs",
      children: ["Public Outputs", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#public-outputs",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Photo hash (SHA-256)"
      }), "\n", jsx(_components.li, {
        children: "Apple Root CA certificate"
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "These committed values allow verifiers to confirm the proof corresponds to a specific photo and chains to Apple's actual root certificate."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
